



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/datasets/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.1">
    
    
      
        <title>Build Datasets - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.30686662.css">
      
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700|Ubuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
        
<script>
  window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-104199698-2", "bactopia.github.io")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })
</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#build-datasets" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bactopia
            </span>
            <span class="md-header-nav__topic">
              
                Build Datasets
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href=".." title="Bactopia" class="md-tabs__link md-tabs__link--active">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../bactopia-tools/" title="Bactopia Tools" class="md-tabs__link">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1" checked>
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Build Datasets
      </label>
    
    <a href="./" title="Build Datasets" class="md-nav__link md-nav__link--active">
      Build Datasets
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#included-datasets" title="Included Datasets" class="md-nav__link">
    Included Datasets
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general" title="General" class="md-nav__link">
    General
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#species-specific" title="Species Specific" class="md-nav__link">
    Species Specific
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up" title="Setting Up" class="md-nav__link">
    Setting Up
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quick-start" title="Quick Start" class="md-nav__link">
    Quick Start
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#a-single-bacterial-species" title="A Single Bacterial Species" class="md-nav__link">
    A Single Bacterial Species
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiple-bacterial-species" title="Multiple Bacterial Species" class="md-nav__link">
    Multiple Bacterial Species
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comma-separated" title="Comma-Separated" class="md-nav__link">
    Comma-Separated
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#text-file" title="Text File" class="md-nav__link">
    Text File
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" title="Usage" class="md-nav__link">
    Usage
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#useful-parameters" title="Useful Parameters" class="md-nav__link">
    Useful Parameters
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#-clear_cache" title="--clear_cache" class="md-nav__link">
    --clear_cache
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-cpus" title="--cpus" class="md-nav__link">
    --cpus
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-force" title="--force*" class="md-nav__link">
    --force*
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-include_genus" title="--include_genus" class="md-nav__link">
    --include_genus
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-limit" title="--limit" class="md-nav__link">
    --limit
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-accessions" title="--accessions" class="md-nav__link">
    --accessions
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-keep_files" title="--keep_files" class="md-nav__link">
    --keep_files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tweaking-cd-hit" title="Tweaking CD-HIT" class="md-nav__link">
    Tweaking CD-HIT
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#datasets-folder-overview" title="Datasets Folder Overview" class="md-nav__link">
    Datasets Folder Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general-datasets" title="General Datasets" class="md-nav__link">
    General Datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#species-specific-datasets" title="Species Specific Datasets" class="md-nav__link">
    Species Specific Datasets
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#annotation" title="annotation" class="md-nav__link">
    annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer" title="minmer" class="md-nav__link">
    minmer
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mlst" title="mlst" class="md-nav__link">
    mlst
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional" title="optional" class="md-nav__link">
    optional
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast" title="blast" class="md-nav__link">
    blast
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mapping-sequences" title="mapping-sequences" class="md-nav__link">
    mapping-sequences
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reference-genomes" title="reference-genomes" class="md-nav__link">
    reference-genomes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../workflow-overview/" title="Workflow Overview" class="md-nav__link">
      Workflow Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/eggnog/" title="eggnog" class="md-nav__link">
      eggnog
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/gtdb/" title="gtdb" class="md-nav__link">
      gtdb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/ismapper/" title="ismapper" class="md-nav__link">
      ismapper
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/mashtree/" title="mashtree" class="md-nav__link">
      mashtree
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/pirate/" title="pirate" class="md-nav__link">
      pirate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/roary/" title="roary" class="md-nav__link">
      roary
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#included-datasets" title="Included Datasets" class="md-nav__link">
    Included Datasets
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general" title="General" class="md-nav__link">
    General
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#species-specific" title="Species Specific" class="md-nav__link">
    Species Specific
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up" title="Setting Up" class="md-nav__link">
    Setting Up
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quick-start" title="Quick Start" class="md-nav__link">
    Quick Start
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#a-single-bacterial-species" title="A Single Bacterial Species" class="md-nav__link">
    A Single Bacterial Species
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiple-bacterial-species" title="Multiple Bacterial Species" class="md-nav__link">
    Multiple Bacterial Species
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comma-separated" title="Comma-Separated" class="md-nav__link">
    Comma-Separated
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#text-file" title="Text File" class="md-nav__link">
    Text File
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" title="Usage" class="md-nav__link">
    Usage
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#useful-parameters" title="Useful Parameters" class="md-nav__link">
    Useful Parameters
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#-clear_cache" title="--clear_cache" class="md-nav__link">
    --clear_cache
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-cpus" title="--cpus" class="md-nav__link">
    --cpus
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-force" title="--force*" class="md-nav__link">
    --force*
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-include_genus" title="--include_genus" class="md-nav__link">
    --include_genus
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-limit" title="--limit" class="md-nav__link">
    --limit
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-accessions" title="--accessions" class="md-nav__link">
    --accessions
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-keep_files" title="--keep_files" class="md-nav__link">
    --keep_files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tweaking-cd-hit" title="Tweaking CD-HIT" class="md-nav__link">
    Tweaking CD-HIT
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#datasets-folder-overview" title="Datasets Folder Overview" class="md-nav__link">
    Datasets Folder Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general-datasets" title="General Datasets" class="md-nav__link">
    General Datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#species-specific-datasets" title="Species Specific Datasets" class="md-nav__link">
    Species Specific Datasets
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#annotation" title="annotation" class="md-nav__link">
    annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer" title="minmer" class="md-nav__link">
    minmer
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mlst" title="mlst" class="md-nav__link">
    mlst
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional" title="optional" class="md-nav__link">
    optional
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast" title="blast" class="md-nav__link">
    blast
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mapping-sequences" title="mapping-sequences" class="md-nav__link">
    mapping-sequences
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reference-genomes" title="reference-genomes" class="md-nav__link">
    reference-genomes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/datasets.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="build-datasets">Build Datasets<a class="headerlink" href="#build-datasets" title="Permanent link">&para;</a></h1>
<p>Bactopia can make use of many existing public datasets, as well as private datasets. The process of downloading, building, and (or) configuring these datasets for Bactopia has been automated.</p>
<div class="admonition info">
<p class="admonition-title">Highly recommended to complete this step!</p>
<p>This step is completely optional, but it is highly recommended that you do not. By skipping this step of setting up public datasets, Bactopia will be limited to analyses like quality control, assembly, and 31-mer counting. </p>
</div>
<h2 id="included-datasets">Included Datasets<a class="headerlink" href="#included-datasets" title="Permanent link">&para;</a></h2>
<p>Some datasets included are applicable to all bacterial species and some are specific to a bacterial species. If specified at runtime, Bactopia will recognize the datasets and execute the appropriate analyses.</p>
<h3 id="general">General<a class="headerlink" href="#general" title="Permanent link">&para;</a></h3>
<p><strong><a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-getref">Ariba's <code>getref</code> Reference Datasets</a></strong><br />
Allows reference datasets (resistance, virulence, and plamids) to be automatically downloaded and configured for usage by Ariba  </p>
<p><strong><a href="https://mash.readthedocs.io/en/latest/data.html">RefSeq Mash Sketch</a></strong><br />
~100,000 genomes and plasmids from NCBI RefSeq, used to give an idea of what is your sequencing data (e.g. Are the sequences what you expected?)  </p>
<p><strong><a href="https://sourmash.readthedocs.io/en/latest/datasets.html?highlight=--track-abundance#genbank-lca-dataset">GenBank Sourmash Signatures</a></strong><br />
~87,000 microbial genomes (includes viral and fungal) from NCBI GenBank, also gives an idea of what is your sequencing data.  </p>
<p><strong><a href="https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/">PLSDB Mash Sketch &amp; BLAST</a></strong><br />
Includes meta data/annotations, Mash sketches, and BLAST database files of all plasmids stored in PLSDB.  </p>
<h3 id="species-specific">Species Specific<a class="headerlink" href="#species-specific" title="Permanent link">&para;</a></h3>
<p><strong><a href="https://pubmlst.org/datasets/">PubMLST.org MLST Schemas</a></strong><br />
Multi-locus sequence typing (MLST) allelic profiles and seqeunces for a many different bacterial species (and even a few eukaryotes!).  </p>
<p><strong>Clustered RefSeq Proteins</strong><br />
For the given bacterial species, completed RefSeq genomes are downloaded and then the proteins are clustered and formatted for usage with Prokka.  </p>
<p><strong>Minmer Sketch of RefSeq Genomes</strong><br />
Using the completed genomes downloaded for clustering proteins a Mash sketch and Sourmash signature is created for these genomes. These sketches can then be used for automatic selection of reference genomes for variant calling.</p>
<p><strong>Optional User Populated Folders</strong><br />
A few folders for things such as calling variants, insertion sequences and primers are created that the user can manually populate. More information is available below!</p>
<h2 id="setting-up">Setting Up<a class="headerlink" href="#setting-up" title="Permanent link">&para;</a></h2>
<p>Included in Bactopia is the <code>setup-datasets.py</code> script (located in the <code>bin</code> folder) to automate the process of downloading and/or building these datasets.</p>
<h3 id="quick-start">Quick Start<a class="headerlink" href="#quick-start" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span>bactopia datasets datasets/
</pre></div>

<p>This will set up Ariba datasets (<code>card</code> and <code>vfdb_core</code>), RefSeq Mash sketch, GenBank Sourmash Signatures, and PLSDB in the newly created <code>datasets</code> folder.</p>
<h3 id="a-single-bacterial-species">A Single Bacterial Species<a class="headerlink" href="#a-single-bacterial-species" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span>bactopia datasets datasets/ --species <span class="s2">&quot;Haemophilus influenzae&quot;</span> --include_genus
</pre></div>

<h3 id="multiple-bacterial-species">Multiple Bacterial Species<a class="headerlink" href="#multiple-bacterial-species" title="Permanent link">&para;</a></h3>
<p>You can also set up datasets for multiple bacterial species at a time. There are two options to do so.</p>
<h4 id="comma-separated">Comma-Separated<a class="headerlink" href="#comma-separated" title="Permanent link">&para;</a></h4>
<p>At runtime, you can separate the the different species
<div class="highlight"><pre><span></span>bactopia datasets datasets/ --species <span class="s2">&quot;Haemophilus influenzae,Staphylococcus aureus&quot;</span> --include_genus
</pre></div></p>
<h4 id="text-file">Text File<a class="headerlink" href="#text-file" title="Permanent link">&para;</a></h4>
<p>In order to do so, you will need to create a text file where each line is the name of a species to set up.</p>
<p>For example, you could create a <code>species.txt</code> file and include the following species in it.
<div class="highlight"><pre><span></span>Haemophilus influenzae
Staphylococcus aureus
Mycobacterium tuberculosis
</pre></div></p>
<p>The new command becomes:</p>
<div class="highlight"><pre><span></span>bactopia datasets datasets/ --species species.txt --include_genus
</pre></div>

<p>This will setup the MLST schema (if available) and a protein cluster FASTA file for each species in <code>species.txt</code>. </p>
<h2 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span>usage: bactopia datasets [-h] [--ariba STR] [--species STR] [--skip_prokka]
                         [--include_genus] [--limit INT] [--accessions STR]
                         [--identity FLOAT] [--overlap FLOAT]
                         [--max_memory INT] [--fast_cluster] [--skip_minmer]
                         [--skip_plsdb] [--cpus INT] [--clear_cache] [--force]
                         [--force_ariba] [--force_mlst] [--force_prokka]
                         [--force_minmer] [--force_plsdb] [--keep_files]
                         [--list_datasets] [--depends] [--version] [--verbose]
                         [--silent]
                         OUTPUT_DIRECTORY

bactopia datasets - Setup public datasets for Bactopia

positional arguments:
  OUTPUT_DIRECTORY  Directory to write output.

optional arguments:
  -h, --help        show this help message and exit

Ariba Reference Datasets:
  --ariba STR       Setup Ariba datasets for a given reference or a list of
                    references in a text file. (Default: card,vfdb_core)

Bacterial Species:
  --species STR     Download available (cg)MLST schemas and completed genomes
                    for a given species or a list of species in a text file.

Custom Prokka Protein FASTA:
  --skip_prokka     Skip creation of a Prokka formatted fasta for each species
  --include_genus   Include all genus members in the Prokka proteins FASTA
  --limit INT       If available completed genomes exceeds a given limit, a
                    random subsample will be taken.
  --accessions STR  A list of RefSeq accessions to download.
  --identity FLOAT  CD-HIT (-c) sequence identity threshold. (Default: 0.9)
  --overlap FLOAT   CD-HIT (-s) length difference cutoff. (Default: 0.8)
  --max_memory INT  CD-HIT (-M) memory limit (in MB). (Default: unlimited
  --fast_cluster    Use CD-HIT&#39;s (-g 0) fast clustering algorithm, instead of
                    the accurate but slow algorithm.

Minmer Datasets:
  --skip_minmer     Skip download of pre-computed minmer datasets (mash,
                    sourmash)

PLSDB (Plasmid) BLAST/Sketch:
  --skip_plsdb      Skip download of pre-computed PLSDB datbases (blast, mash)

Helpful Options:
  --cpus INT        Number of cpus to use. (Default: 1)
  --clear_cache     Remove any existing cache.
  --force           Forcibly overwrite existing datasets.
  --force_ariba     Forcibly overwrite existing Ariba datasets.
  --force_mlst      Forcibly overwrite existing MLST datasets.
  --force_prokka    Forcibly overwrite existing Prokka datasets.
  --force_minmer    Forcibly overwrite existing minmer datasets.
  --force_plsdb     Forcibly overwrite existing PLSDB datasets.
  --keep_files      Keep all downloaded and intermediate files.
  --list_datasets   List Ariba reference datasets and (cg)MLST schemas
                    available for setup.
  --depends         Verify dependencies are installed.

Adjust Verbosity:
  --version         show program&#39;s version number and exit
  --verbose         Print debug related text.
  --silent          Only critical errors will be printed.

example usage:
  setup-public-datasets.py outdir
  setup-public-datasets.py outdir --ariba &#39;card&#39;
  setup-public-datasets.py outdir --species &#39;Staphylococcus aureus&#39; --include_genus
</pre></div>

<h3 id="useful-parameters">Useful Parameters<a class="headerlink" href="#useful-parameters" title="Permanent link">&para;</a></h3>
<h4 id="-clear_cache">--clear_cache<a class="headerlink" href="#-clear_cache" title="Permanent link">&para;</a></h4>
<p>To determine which MLST schemas are available, PubMLST.org is queryied. To prevent a query every run, a list of available schemas is cached to <code>$HOME/.bactopia/datasets.json</code>. The cache expires after 15 days, but in case a new species has been made available <code>--clear_cache</code> will force a requery of PubMLST.org.</p>
<h4 id="-cpus">--cpus<a class="headerlink" href="#-cpus" title="Permanent link">&para;</a></h4>
<p>Increasing <code>--cpus</code> (it defaults to 1) is useful for speeding up the download and clustering steps.</p>
<h4 id="-force">--force*<a class="headerlink" href="#-force" title="Permanent link">&para;</a></h4>
<p>If a dataset exists, it will only be overwritten if one of the <code>--force</code> parameters are used.</p>
<h4 id="-include_genus">--include_genus<a class="headerlink" href="#-include_genus" title="Permanent link">&para;</a></h4>
<p>Completed RefSeq genomes are downloaded for a given species to be used for protein clustering. <code>--include_genus</code> will also download completed RefSeq genomes for each genus member.</p>
<h4 id="-limit">--limit<a class="headerlink" href="#-limit" title="Permanent link">&para;</a></h4>
<p>For some species of bacteria there might be thousands of completed genomes available. For dataset creation, downloading thousands of completed genomes will be time consuming and like take up a significant amount of storage. To help in such cases <code>--limit</code> can be used to limit the downloads to a random subset of genomes.</p>
<h4 id="-accessions">--accessions<a class="headerlink" href="#-accessions" title="Permanent link">&para;</a></h4>
<p>In cases where a random subset of completed genomes is not ideal, you can provide your own curated list of genomes to download with <code>--accessions</code>. The file should have a single NCBI RefSeq Assembly accession (E.g GCF_000008865.2) per line.</p>
<h4 id="-keep_files">--keep_files<a class="headerlink" href="#-keep_files" title="Permanent link">&para;</a></h4>
<p>Many intermediate files are downloaded/created (e.g. completed genomes) and deleted during the building process, use <code>--keep_files</code> to retain these files.</p>
<h4 id="tweaking-cd-hit">Tweaking CD-HIT<a class="headerlink" href="#tweaking-cd-hit" title="Permanent link">&para;</a></h4>
<p>There are parameters (<code>--identity</code>, <code>--overlap</code>, <code>--max_memory</code>, and <code>--fast_cluster</code>) to tweak CD-HIT if you find it necessary. Please keep in mind, the only goal of the protein clustering step is to help speed up Prokka, by providing a decent set of proteins to annotate against first.</p>
<h2 id="datasets-folder-overview">Datasets Folder Overview<a class="headerlink" href="#datasets-folder-overview" title="Permanent link">&para;</a></h2>
<p>After creating datasets you will have a directory structure that Bactopia recognizes. Based on the available datasets Bactopia will queue up the associated analyses.</p>
<p>Here is the directory structure for the Bactopia Datasets. Some of these include files from public datasets that can be used directly, but there are also other folders you can populate yourself to fit your needs.
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
├── ariba
├── minmer
├── plasmid
└── species-specific
    └── ${SPECIES}
        ├── annotation
        │   ├── cdhit-stats.txt
        │   ├── genome_size.json
        │   ├── ncbi-metadata.txt
        │   ├── proteins.faa
        │   ├── proteins.faa.clstr
        │   └── proteins-updated.txt
        ├── minmer
        │   ├── minmer-updated.txt
        │   └── refseq-genomes.msh
        ├── mlst
        │   └── ${SCHEMA}
        │       ├── ariba.tar.gz
        │       ├── blastdb.tar.gz
        │       └── mlst-updated.txt
        └── optional
            ├── blast
            │   ├── genes
            │   │   └── ${NAME}.fasta
            │   ├── primers
            │   │   └── ${NAME}.fasta
            │   └── proteins
            │       └── ${NAME}.fasta
            ├── insertion-sequences
            │   └── ${NAME}.fasta
            ├── mapping-sequences
            │   └── ${NAME}.fasta
            └── reference-genomes
                └── ${NAME}.{gbk|fasta}
</pre></div></p>
<h3 id="general-datasets">General Datasets<a class="headerlink" href="#general-datasets" title="Permanent link">&para;</a></h3>
<p>General datasets can be used for all bacterial samples. There are three general dataset folders: <code>ariba</code>, <code>minmer</code> and <code>plasmid</code>.</p>
<p>The <code>ariba</code> folder contains pre-formatted datasets available from <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-getref">Ariba's <code>getref</code> Reference Datasets</a>. </p>
<p>The <code>minmer</code> folder contains a <a href="https://mash.readthedocs.io/en/latest/data.html">RefSeq Mash Sketch</a> and <a href="https://sourmash.readthedocs.io/en/latest/datasets.html?highlight=--track-abundance#genbank-lca-dataset">GenBank Sourmash Signatures</a> of more than 100,000 genomes.</p>
<p>Finally, the <code>plasmid</code> folder contains a <a href="https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/">PLSDB Mash Sketch &amp; BLAST database</a> from a curated set of plasmids.</p>
<div class="admonition warning">
<p class="admonition-title">Changing files in <code>ariba</code>, <code>minmer</code> and <code>plasmid</code> is not recommended</p>
<p>These directories are for general analysis and have been precomputed. Modifying these files
may cause errors during analysis.</p>
</div>
<h3 id="species-specific-datasets">Species Specific Datasets<a class="headerlink" href="#species-specific-datasets" title="Permanent link">&para;</a></h3>
<p>Bactopia allows the datasets to be created for a specific species. The following sections outline the species specific datasets.</p>
<h4 id="annotation">annotation<a class="headerlink" href="#annotation" title="Permanent link">&para;</a></h4>
<p>Completed RefSeq genomes are downloaded and then the proteins are clustered and formatted for usage with Prokka. The results from this clustering is stored in the <code>annotation</code> folder.<br />
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
└── species-specific
    └── ${SPECIES}
        └── annotation
            ├── cdhit-stats.txt
            ├── genome_size.json
            ├── ncbi-metadata.txt
            ├── proteins.faa
            ├── proteins.faa.clstr
            └── proteins-updated.txt
</pre></div></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>cdhit-stats.txt</td>
<td>General statistics associated with CD-HIT clustering</td>
</tr>
<tr>
<td>genome_size.json</td>
<td>A list of genome size for each downloaded RefSeq genome</td>
</tr>
<tr>
<td>ncbi-metadata.txt</td>
<td>NCBI Assembly metadata associated with the downloaded RefSeq genomes</td>
</tr>
<tr>
<td>proteins.faa</td>
<td>Set of <a href="https://github.com/tseemann/prokka#fasta-database-format">Prokka formatted proteins</a></td>
</tr>
<tr>
<td>proteins.faa.clstr</td>
<td>Description of the clusters created by CD-HIT</td>
</tr>
<tr>
<td>proteins-updated.txt</td>
<td>Information on the last time the protein set was updated</td>
</tr>
</tbody>
</table>
<div class="admonition info">
<p class="admonition-title">You can add your curated protein set here</p>
<p>If you have a set of proteins you would like to use for annotation, you can name it <code>proteins.faa</code> and place it 
in the <code>annotation</code> folder. In order for your set of proteins to be used by Prokka, you must make sure you follow
the <a href="https://github.com/tseemann/prokka#fasta-database-format">Prokka FASTA database format</a>.</p>
<p>An alternative is to use the <code>--accessions</code> parameter and give <code>bactopia datasets</code> the list of RefSeq accessions when the
dataset is created. In doing so the custom protein set will be automatically formatted using the genomes you specified.</p>
</div>
<h4 id="minmer">minmer<a class="headerlink" href="#minmer" title="Permanent link">&para;</a></h4>
<p>By default, a <a href="https://github.com/marbl/Mash">Mash sketch</a> is created for the completed genomes downloaded for clustering proteins. These sketches are then be used for automatic selection of reference genomes for variant calling.
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
└── species-specific
    └── ${SPECIES}
        └── minmer
            ├── minmer-updated.txt
            └── refseq-genomes.msh
</pre></div>
| Filename | Description |
|----------|-------------|
| minmer-updated.txt | Information on the last time the mash sketch was updated |
| refseq-genomes.msh | A Mash sketch (k=31) of the RefSeq completed genomes |</p>
<div class="admonition info">
<p class="admonition-title">You can add your curated RefSeq sketch here</p>
<p>You can replace <code>refseq-genomes.msh</code> with a custom set of RefSeq genomes to be used for automatic reference selection. 
The only requirements to do so are that only RefSeq genomes (start with GCF) are used and the <code>mash sketch</code> uses a k-mer
length of 31 (<code>-k 31</code>). This will allow it to be compatible with Bactopia.</p>
<p>An alternative is to use the <code>--accessions</code> parameter and give <code>bactopia datasets</code> the list of RefSeq accessions when the
dataset is created. In doing so the mash sketch will be automatically created.</p>
</div>
<h4 id="mlst">mlst<a class="headerlink" href="#mlst" title="Permanent link">&para;</a></h4>
<p>The <code>mlst</code> folder contains MLST schemas that have been formatted to be used by <a href="https://github.com/sanger-pathogens/ariba/">Ariba</a> and <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a>.
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
└── species-specific
    └── ${SPECIES}
        └── mlst
            └── ${SCHEMA}
                ├── ariba.tar.gz
                ├── blastdb.tar.gz
                └── mlst-updated.txt
</pre></div></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>ariba.tar.gz</td>
<td>An Ariba formatted MLST dataset for a given schema</td>
</tr>
<tr>
<td>blastdb.tar.gz</td>
<td>A BLAST formatted MLST dataset for a given schema</td>
</tr>
<tr>
<td>mlst-updated.txt</td>
<td>Contains time stamp for the last time the MSLT dataset was updated</td>
</tr>
</tbody>
</table>
<div class="admonition info">
<p class="admonition-title">How does Bactopia handle organisms with multiple MLST schemas?</p>
<p>In a few cases, an organism might have multiple MLST schemas available (Example: <a href="https://pubmlst.org/escherichia/">E. coli</a>). In such cases, each MLST schema is downloaded and set up. Bactopia will also call sequence types against each schema. </p>
</div>
<div class="admonition warning">
<p class="admonition-title">Changing files in <code>mlst</code> is not recommended</p>
<p>The MLST schemas have been pre-formatted for your usage. There might be rare cases where you would like to provide your own schema. If this is the case it is recommended you take a look at: <a href="https://github.com/sanger-pathogens/ariba/issues/185">What about MLST not hosted at pubmlst.org?</a> then follow the directory structure for <code>mlst</code>.</p>
</div>
<h4 id="optional">optional<a class="headerlink" href="#optional" title="Permanent link">&para;</a></h4>
<p>Built into the Bactopia dataset structure is the <code>optional</code> folder that you, the user, can populate for species specific analysis. These could include specific genes you might want BLASTed against your samples or a specific reference you want all your samples mapped to and variants called.</p>
<h5 id="blast">blast<a class="headerlink" href="#blast" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
└── species-specific
    └── ${SPECIES}
        └── optional
            └── blast
                ├── genes
                │   └── ${NAME}.fasta
                ├── primers
                │   └── ${NAME}.fasta
                └── proteins
                    └── ${NAME}.fasta
</pre></div>

<p>In the <code>blast</code> directory there are three more directories! </p>
<p>The <code>genes</code> folder is where you can place gene seqeunces (nucleotides) in FASTA format to query against assemblies using <code>blastn</code>.</p>
<p>The <code>primers</code> folder is where you can place primer sequences (nucleotides) in FASTA format to query against assemblies using <code>blastn</code>, but with primer-specific parameters and cut-offs.</p>
<p>Finally, the <code>proteins</code> (as you probably guessed!) is where you can place protein sequnces (amino acids) in FASTA format to query against assemblies using <code>blastp</code>.</p>
<h5 id="mapping-sequences">mapping-sequences<a class="headerlink" href="#mapping-sequences" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
└── species-specific
    └── ${SPECIES}
        └── optional
            └── mapping-sequences
                └── ${NAME}.fasta
</pre></div>

<p>In the <code>mapping-sequences</code> directory you can place FASTA files of any nucleotide sequence you would like FASTQ reads to be mapped against using <a href="https://github.com/lh3/bwa">BWA</a>. This can be useful if you are interested if whether a certain region or gene is covered or not.</p>
<h5 id="reference-genomes">reference-genomes<a class="headerlink" href="#reference-genomes" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span>${DATASET_FOLDER}
└── species-specific
    └── ${SPECIES}
        └── optional
            └── reference-genomes
                └── ${NAME}.{gbk|fasta}
</pre></div>

<p>In the <code>reference-genomes</code> directory you can put a GenBank (preferred!) or FASTA file of a reference genome you would like variants to be called against using <a href="https://github.com/tseemann/snippy">Snippy</a>.</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../tutorial/" title="Tutorial" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Tutorial
              </span>
            </div>
          </a>
        
        
          <a href="../usage-basic/" title="Basic Usage" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Basic Usage
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright © 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.c648116f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>