



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/bactopia-tools/gtdb/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.1">
    
    
      
        <title>gtdb - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.30686662.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700|Ubuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
    
      
        
<script>
  window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-104199698-2", "bactopia.github.io")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })
</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#bactopia-tools-gtdb" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bactopia
            </span>
            <span class="md-header-nav__topic">
              
                gtdb
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../.." title="Bactopia" class="md-tabs__link">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../" title="Bactopia Tools" class="md-tabs__link md-tabs__link--active">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../.." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../datasets/" title="Build Datasets" class="md-nav__link">
      Build Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../workflow-overview/" title="Workflow Overview" class="md-nav__link">
      Workflow Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        gtdb
      </label>
    
    <a href="./" title="gtdb" class="md-nav__link md-nav__link--active">
      gtdb
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" title="Example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" title="Output Overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" title="Directory Description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align" title="align" class="md-nav__link">
    align
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bactopia-info" title="bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#classify" title="classify" class="md-nav__link">
    classify
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identify" title="identify" class="md-nav__link">
    identify
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#completed-usage" title="Completed Usage" class="md-nav__link">
    Completed Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary/" title="roary" class="md-nav__link">
      roary
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" title="Example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" title="Output Overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" title="Directory Description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align" title="align" class="md-nav__link">
    align
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bactopia-info" title="bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#classify" title="classify" class="md-nav__link">
    classify
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identify" title="identify" class="md-nav__link">
    identify
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#completed-usage" title="Completed Usage" class="md-nav__link">
    Completed Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/bactopia-tools/gtdb.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="bactopia-tools-gtdb">Bactopia Tools - <em>gtdb</em><a class="headerlink" href="#bactopia-tools-gtdb" title="Permanent link">&para;</a></h1>
<p>The <code>gtdb</code> tool uses <a href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk's</a> classify 
workflow to assign taxonomic classifications to your set of samples. This is 
done through the use of the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a>. 
If you are unsure of your sequences, <code>gtdb</code> is useful tool to help determine
the taxonomy of your samples.</p>
<h2 id="example">Example<a class="headerlink" href="#example" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span>bactopia tools gtdb \
    --bactopia ~/bactopia-tutorial/bactopia \
    --gtdb ~/bactopia-tutorial/bactopia-datasets/gtdb/db \
    --exclude ~/bactopia-tutorial/bactopia-tools/summary/bactopia-exclude.txt \
    --cpus 4
</pre></div>

<h2 id="output-overview">Output Overview<a class="headerlink" href="#output-overview" title="Permanent link">&para;</a></h2>
<p>Below is the default output structure for the <code>gtdb</code> tool. Where possible the 
file descriptions below were modified from descriptions at the 
<a href="https://github.com/Ecogenomics/GTDBTk#classify-workflow">GTDB-Tk Classify Workflow</a> page.</p>
<div class="highlight"><pre><span></span>bactopia-tools/
└── gtdb
    ├── bactopia-info
    │   ├── gtdb-report.html
    │   ├── gtdb-timeline.html
    │   └── gtdb-trace.txt
    └── classify
        ├── align
        │   ├── gtdb.(ar122|bac120).filtered.tsv
        │   ├── gtdb.(ar122|bac120).msa.fasta
        │   ├── gtdb.(ar122|bac120).user_msa.fasta
        │   └── intermediate_results
        │       └── gtdb.(ar122|bac120).marker_info.tsv
        ├── classify
        │   ├── gtdb.(ar122|bac120).classify.tree
        │   ├── gtdb.(ar122|bac120).summary.tsv
        │   └── intermediate_results
        │       ├── gtdb.(ar122|bac120).classification_pplacer.tsv
        │       ├── gtdb.(ar122|bac120).red_dictionary.tsv
        │       └── pplacer
        │           ├── pplacer.(ar122|bac120).json
        │           └── pplacer.(ar122|bac120).out
        ├── gtdb.(ar122|bac120).classify.tree
        ├── gtdb.(ar122|bac120).filtered.tsv
        ├── gtdb.(ar122|bac120).markers_summary.tsv
        ├── gtdb.(ar122|bac120).msa.fasta
        ├── gtdb.(ar122|bac120).summary.tsv
        ├── gtdb.(ar122|bac120).user_msa.fasta
        ├── gtdbtk.log
        ├── gtdbtk.warnings.log
        ├── gtdb.translation_table_summary.tsv
        └── identify
            ├── gtdb.ar122.markers_summary.tsv
            ├── gtdb.bac120.markers_summary.tsv
            ├── gtdb.translation_table_summary.tsv
            └── intermediate_results
                └── marker_genes
                    └── ${SAMPLE_NAME}
                        ├── prodigal_translation_table.tsv
                        ├── prodigal_translation_table.tsv.sha256
                        ├── ${SAMPLE_NAME}_pfam_tophit.tsv
                        ├── ${SAMPLE_NAME}_pfam_tophit.tsv.sha256
                        ├── ${SAMPLE_NAME}_pfam.tsv
                        ├── ${SAMPLE_NAME}_pfam.tsv.sha256
                        ├── ${SAMPLE_NAME}_protein.faa
                        ├── ${SAMPLE_NAME}_protein.faa.sha256
                        ├── ${SAMPLE_NAME}_protein.fna
                        ├── ${SAMPLE_NAME}_protein.fna.sha256
                        ├── ${SAMPLE_NAME}_protein.gff
                        ├── ${SAMPLE_NAME}_protein.gff.sha256
                        ├── ${SAMPLE_NAME}_tigrfam.out
                        ├── ${SAMPLE_NAME}_tigrfam.out.sha256
                        ├── ${SAMPLE_NAME}_tigrfam_tophit.tsv
                        ├── ${SAMPLE_NAME}_tigrfam_tophit.tsv.sha256
                        ├── ${SAMPLE_NAME}_tigrfam.tsv
                        └── ${SAMPLE_NAME}_tigrfam.tsv.sha256
</pre></div>

<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.(ar122|bac120).classify.tree</td>
<td>Reference tree in Newick format containing query genomes placed with pplacer</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).filtered.tsv</td>
<td>List of genomes with an insufficient number of amino acids in MSA</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).markers_summary.tsv</td>
<td>Markers used in generation of the concatenated MSA and the order in which they were applied</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).msa.fasta</td>
<td>FASTA file containing MSA of submitted and reference genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).summary.tsv</td>
<td>A summary of classifications provided by GTDB-Tk, see <a href="https://github.com/Ecogenomics/GTDBTk#classification-summary-file">classification summary</a> for more details</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).user_msa.fasta</td>
<td>FASTA file containing MSA of the submitted genomes</td>
</tr>
<tr>
<td>gtdbtk.log</td>
<td>A log of the run</td>
</tr>
<tr>
<td>gtdbtk.warnings.log</td>
<td>A log of any warnings produced by the run</td>
</tr>
<tr>
<td>gtdb.translation_table_summary.tsv</td>
<td>Summary of the tranlastlation table used for each genome</td>
</tr>
</tbody>
</table>
<h3 id="directory-description">Directory Description<a class="headerlink" href="#directory-description" title="Permanent link">&para;</a></h3>
<h4 id="align">align<a class="headerlink" href="#align" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.(ar122|bac120).filtered.tsv</td>
<td>List of genomes with an insufficient number of amino acids in MSA</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).msa.fasta</td>
<td>FASTA file containing MSA of submitted and reference genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).user_msa.fasta</td>
<td>FASTA file containing MSA of the submitted genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).marker_info.tsv</td>
<td>Markers used in generation of the concatenated MSA and the order in which they were applied</td>
</tr>
</tbody>
</table>
<h4 id="bactopia-info">bactopia-info<a class="headerlink" href="#bactopia-info" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb-report.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#execution-report">Execution Report</a></td>
</tr>
<tr>
<td>gtdb-timeline.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#timeline-report">Timeline Report</a></td>
</tr>
<tr>
<td>gtdb-trace.txt</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">Trace</a> report</td>
</tr>
</tbody>
</table>
<h4 id="classify">classify<a class="headerlink" href="#classify" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.(ar122|bac120).classify.tree</td>
<td>Reference tree in Newick format containing query genomes placed with pplacer</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).summary.tsv</td>
<td>Classification of query genomes based on their placement in the reference tree, relative evolutionary divergence, and ANI to reference genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).classification_pplacer.tsv</td>
<td>Classification of query genomes based only on their placement in the reference tree</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).red_dictionary.tsv</td>
<td>Median RED values for taxonomic ranks</td>
</tr>
<tr>
<td>pplacer.(ar122|bac120).json</td>
<td>Output information generated by pplacer in JSON format</td>
</tr>
<tr>
<td>pplacer.(ar122|bac120).out</td>
<td>Output information generated by pplacer</td>
</tr>
</tbody>
</table>
<h4 id="identify">identify<a class="headerlink" href="#identify" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.ar122.markers_summary.tsv</td>
<td>Summary of unique, duplicated, and missing markers within the 122 archaeal marker set for each submitted genome</td>
</tr>
<tr>
<td>gtdb.bac120.markers_summary.tsv</td>
<td>Summary of unique, duplicated, and missing markers within the 120 bacterial marker set for each submitted genome</td>
</tr>
<tr>
<td>gtdb.translation_table_summary.tsv</td>
<td>The predicted translation table used for gene calling for each genome</td>
</tr>
<tr>
<td>identify/intermediate_results/marker_genes/</td>
<td>Contains individual genome results for gene calling using Prodigal and gene identification based on TIGRFAM and Pfam HMMs</td>
</tr>
</tbody>
</table>
<h2 id="completed-usage">Completed Usage<a class="headerlink" href="#completed-usage" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span>Required Parameters:
    --bactopia STR          Directory containing Bactopia analysis results for all samples.

    --gtdb STR              Location of a GTDB database. If a database is not found, you must
                                use &#39;--download_gtdb&#39;.

Optional Parameters:
    --include STR           A text file containing sample names to include in the
                                analysis. The expected format is a single sample per line.

    --exclude STR           A text file containing sample names to exclude from the
                                analysis. The expected format is a single sample per line.

    --prefix DIR            Prefix to use for final output files
                                Default: gtdb

    --outdir DIR            Directory to write results to
                                Default: ./

    --max_time INT          The maximum number of minutes a job should run before being halted.
                                Default: 120 minutes

    --max_memory INT        The maximum amount of memory (Gb) allowed to a single process.
                                Default: 32 Gb

    --cpus INT              Number of processors made available to a single
                                process.
                                Default: 1

GTDB-Tk Related Parameters:
    --download_gtdb         Download the latest GTDB database, even it exists.

    --min_perc_aa INT       Filter genomes with an insufficient percentage of AA in the MSA
                                Default: 10

    --recalculate_red       Recalculate RED values based on the reference tree and all added
                                user genomes

    --force_gtdb            Force GTDB to continue processing if an error occurrs on a single
                                genome

    --debug                 Create intermediate files for debugging purposes

Nextflow Related Parameters:
    --publish_mode          Set Nextflow&#39;s method for publishing output files. Allowed methods are:
                                &#39;copy&#39; (default)    Copies the output files into the published directory.

                                &#39;copyNoFollow&#39; Copies the output files into the published directory
                                               without following symlinks ie. copies the links themselves.

                                &#39;link&#39;    Creates a hard link in the published directory for each
                                          process output file.

                                &#39;rellink&#39; Creates a relative symbolic link in the published directory
                                          for each process output file.

                                &#39;symlink&#39; Creates an absolute symbolic link in the published directory
                                          for each process output file.

                                Default: copy

    --force                 Nextflow will overwrite existing output files.
                                Default: false

    --conatainerPath        Path to Singularity containers to be used by the &#39;slurm&#39;
                                profile.
                                Default: /opt/bactopia/singularity

    --sleep_time            After reading datases, the amount of time (seconds) Nextflow
                                will wait before execution.
                                Default: 5 seconds

Useful Parameters:
    --version               Print workflow version information
    --help                  Show this message and exit
</pre></div>

<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p><strong><a href="https://github.com/ParBLiSS/FastANI">FastANI</a></strong><br />
<em>Jain, C., Rodriguez-R, L. M., Phillippy, A. M., Konstantinidis, K. T. &amp; Aluru, S. 
<a href="http://dx.doi.org/10.1038/s41467-018-07641-9">High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries.</a>
 Nat. Commun. 9, 5114 (2018)</em>  </p>
</li>
<li>
<p><strong><a href="http://www.microbesonline.org/fasttree">FastTree 2</a></strong><br />
<em>Price, M. N., Dehal, P. S. &amp; Arkin, A. P. 
<a href="https://dx.doi.org/10.1371%2Fjournal.pone.0009490">FastTree 2 – Approximately Maximum-Likelihood Trees for Large Alignments.</a>
 PLoS One 5, e9490 (2010)</em>  </p>
</li>
<li>
<p><strong><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a></strong><br />
_Parks, D. H. et al. 
<a href="https://doi.org/10.1038/nbt.4229">A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life.</a>
 Nat. Biotechnol. 36, 996–1004 (2018) <em><br />
_Parks, D. H. et al. 
<a href="https://doi.org/10.1101/771964">Selection of representative genomes for 24,706 bacterial and archaeal species clusters provide a complete genome-based taxonomy.</a>
 bioRxiv 771964 (2019)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk</a></strong><br />
<em>Chaumeil, P.-A., Mussig, A. J., Hugenholtz, P. &amp; Parks, D. H. 
<a href="https://doi.org/10.1093/bioinformatics/btz848">GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database.</a>
 Bioinformatics (2019)</em>  </p>
</li>
<li>
<p><strong><a href="http://hmmer.org/">HMMER3</a></strong><br />
<em>Eddy, S. R. 
<a href="https://doi.org/10.1371/journal.pcbi.1002195">Accelerated Profile HMM Searches.</a> 
PLoS Comput. Biol. 7, e1002195 (2011)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/matsen/pplacer">pplacer</a></strong><br />
<em>Matsen, F. A., Kodner, R. B. &amp; Armbrust, E. V. 
<a href="https://doi.org/10.1186/1471-2105-11-538">pplacer: linear time maximum-likelihood and Bayesian phylogenetic placement of sequences onto a fixed reference tree.</a>
 BMC Bioinformatics 11, 538 (2010)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/hyattpd/Prodigal">Prodigal</a></strong><br />
<em>Hyatt, D. et al. 
<a href="https://doi.org/10.1186/1471-2105-11-119">Prodigal: prokaryotic gene recognition and translation initiation site identification.</a>
 BMC Bioinformatics 11, 119 (2010)</em>  </p>
</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../fastani/" title="fastani" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                fastani
              </span>
            </div>
          </a>
        
        
          <a href="../phyloflash/" title="phyloflash" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                phyloflash
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright © 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.c648116f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
  </body>
</html>