


<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/bactopia-tools/gtdb/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-5.5.0">
    
    
      
        <title>gtdb - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.b5d04df8.min.css">
      
      
    
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700%7CUbuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
        
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-104199698-2","bactopia.github.io"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}),document.addEventListener("DOMContentSwitch",function(){ga("send","pageview",document.location.pathname)})</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bactopia-tools-gtdb" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo" aria-label="Bactopia">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            Bactopia
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              gtdb
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
        
      
      
        
          

  

<nav class="md-tabs md-tabs--active" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../.." class="md-tabs__link">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../" class="md-tabs__link md-tabs__link--active">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Bactopia" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../.." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../datasets/" title="Build Datasets" class="md-nav__link">
      Build Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../workflow-overview/" title="Workflow Overview" class="md-nav__link">
      Workflow Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Bactopia Tools" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../eggnog/" title="eggnog" class="md-nav__link">
      eggnog
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        gtdb
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 9h14V7H3v2m0 4h14v-2H3v2m0 4h14v-2H3v2m16 0h2v-2h-2v2m0-10v2h2V7h-2m0 6h2v-2h-2v2z"/></svg>
        </span>
      </label>
    
    <a href="./" title="gtdb" class="md-nav__link md-nav__link--active">
      gtdb
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav" aria-label="Output Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav" aria-label="Directory Description">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align" class="md-nav__link">
    align
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#classify" class="md-nav__link">
    classify
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identify" class="md-nav__link">
    identify
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conda-environment" class="md-nav__link">
    Conda Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../ismapper/" title="ismapper" class="md-nav__link">
      ismapper
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mashtree/" title="mashtree" class="md-nav__link">
      mashtree
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pirate/" title="pirate" class="md-nav__link">
      pirate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary/" title="roary" class="md-nav__link">
      roary
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav" aria-label="Output Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav" aria-label="Directory Description">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align" class="md-nav__link">
    align
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#classify" class="md-nav__link">
    classify
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identify" class="md-nav__link">
    identify
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conda-environment" class="md-nav__link">
    Conda Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/bactopia-tools/gtdb.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  
                
                
                <h1 id="bactopia-tools-gtdb">Bactopia Tools - <em>gtdb</em><a class="headerlink" href="#bactopia-tools-gtdb" title="Permanent link">&para;</a></h1>
<p>The <code>gtdb</code> tool uses <a href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk's</a> classify 
workflow to assign taxonomic classifications to your set of samples. This is 
done through the use of the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a>. 
If you are unsure of your sequences, <code>gtdb</code> is useful tool to help determine
the taxonomy of your samples.</p>
<h2 id="example">Example<a class="headerlink" href="#example" title="Permanent link">&para;</a></h2>
<p>The following command will use <code>gtdb</code> to assign a taxonomic classification on all 
samples except those listed in the <em>exclude</em> file.
<div class="highlight"><pre><span></span><code>bactopia tools gtdb \
    --bactopia ~/bactopia-tutorial/bactopia \
    --gtdb ~/bactopia-tutorial/bactopia-datasets/gtdb/db \
    --exclude ~/bactopia-tutorial/bactopia-tools/summary/bactopia-exclude.txt \
    --cpus 4
</code></pre></div></p>
<h2 id="output-overview">Output Overview<a class="headerlink" href="#output-overview" title="Permanent link">&para;</a></h2>
<p>Below is the default output structure for the <code>gtdb</code> tool. Where possible the 
file descriptions below were modified from descriptions at the 
<a href="https://github.com/Ecogenomics/GTDBTk#classify-workflow">GTDB-Tk Classify Workflow</a> page.</p>
<div class="highlight"><pre><span></span><code>bactopia-tools/
└── gtdb
    └── ${PREFIX}
        ├── bactopia-info
        │   ├── gtdb-report.html
        │   ├── gtdb-timeline.html
        │   └── gtdb-trace.txt
        └── classify
            ├── align
            │   ├── gtdb.(ar122|bac120).filtered.tsv
            │   ├── gtdb.(ar122|bac120).msa.fasta
            │   ├── gtdb.(ar122|bac120).user_msa.fasta
            │   └── intermediate_results
            │       └── gtdb.(ar122|bac120).marker_info.tsv
            ├── classify
            │   ├── gtdb.(ar122|bac120).classify.tree
            │   ├── gtdb.(ar122|bac120).summary.tsv
            │   └── intermediate_results
            │       ├── gtdb.(ar122|bac120).classification_pplacer.tsv
            │       ├── gtdb.(ar122|bac120).red_dictionary.tsv
            │       └── pplacer
            │           ├── pplacer.(ar122|bac120).json
            │           └── pplacer.(ar122|bac120).out
            ├── gtdb.(ar122|bac120).classify.tree
            ├── gtdb.(ar122|bac120).filtered.tsv
            ├── gtdb.(ar122|bac120).markers_summary.tsv
            ├── gtdb.(ar122|bac120).msa.fasta
            ├── gtdb.(ar122|bac120).summary.tsv
            ├── gtdb.(ar122|bac120).user_msa.fasta
            ├── gtdbtk.log
            ├── gtdbtk.warnings.log
            ├── gtdb.translation_table_summary.tsv
            └── identify
                ├── gtdb.ar122.markers_summary.tsv
                ├── gtdb.bac120.markers_summary.tsv
                ├── gtdb.translation_table_summary.tsv
                └── intermediate_results
                    └── marker_genes
                        └── ${SAMPLE_NAME}
                            ├── prodigal_translation_table.tsv
                            ├── prodigal_translation_table.tsv.sha256
                            ├── ${SAMPLE_NAME}_pfam_tophit.tsv
                            ├── ${SAMPLE_NAME}_pfam_tophit.tsv.sha256
                            ├── ${SAMPLE_NAME}_pfam.tsv
                            ├── ${SAMPLE_NAME}_pfam.tsv.sha256
                            ├── ${SAMPLE_NAME}_protein.faa
                            ├── ${SAMPLE_NAME}_protein.faa.sha256
                            ├── ${SAMPLE_NAME}_protein.fna
                            ├── ${SAMPLE_NAME}_protein.fna.sha256
                            ├── ${SAMPLE_NAME}_protein.gff
                            ├── ${SAMPLE_NAME}_protein.gff.sha256
                            ├── ${SAMPLE_NAME}_tigrfam.out
                            ├── ${SAMPLE_NAME}_tigrfam.out.sha256
                            ├── ${SAMPLE_NAME}_tigrfam_tophit.tsv
                            ├── ${SAMPLE_NAME}_tigrfam_tophit.tsv.sha256
                            ├── ${SAMPLE_NAME}_tigrfam.tsv
                            └── ${SAMPLE_NAME}_tigrfam.tsv.sha256
</code></pre></div>

<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.(ar122|bac120).classify.tree</td>
<td>Reference tree in Newick format containing query genomes placed with pplacer</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).filtered.tsv</td>
<td>List of genomes with an insufficient number of amino acids in MSA</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).markers_summary.tsv</td>
<td>Markers used in generation of the concatenated MSA and the order in which they were applied</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).msa.fasta</td>
<td>FASTA file containing MSA of submitted and reference genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).summary.tsv</td>
<td>A summary of classifications provided by GTDB-Tk, see <a href="https://github.com/Ecogenomics/GTDBTk#classification-summary-file">classification summary</a> for more details</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).user_msa.fasta</td>
<td>FASTA file containing MSA of the submitted genomes</td>
</tr>
<tr>
<td>gtdbtk.log</td>
<td>A log of the run</td>
</tr>
<tr>
<td>gtdbtk.warnings.log</td>
<td>A log of any warnings produced by the run</td>
</tr>
<tr>
<td>gtdb.translation_table_summary.tsv</td>
<td>Summary of the tranlastlation table used for each genome</td>
</tr>
</tbody>
</table>
<h3 id="directory-description">Directory Description<a class="headerlink" href="#directory-description" title="Permanent link">&para;</a></h3>
<h4 id="align">align<a class="headerlink" href="#align" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.(ar122|bac120).filtered.tsv</td>
<td>List of genomes with an insufficient number of amino acids in MSA</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).msa.fasta</td>
<td>FASTA file containing MSA of submitted and reference genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).user_msa.fasta</td>
<td>FASTA file containing MSA of the submitted genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).marker_info.tsv</td>
<td>Markers used in generation of the concatenated MSA and the order in which they were applied</td>
</tr>
</tbody>
</table>
<h4 id="bactopia-info">bactopia-info<a class="headerlink" href="#bactopia-info" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb-report.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#execution-report">Execution Report</a></td>
</tr>
<tr>
<td>gtdb-timeline.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#timeline-report">Timeline Report</a></td>
</tr>
<tr>
<td>gtdb-trace.txt</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">Trace</a> report</td>
</tr>
</tbody>
</table>
<h4 id="classify">classify<a class="headerlink" href="#classify" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.(ar122|bac120).classify.tree</td>
<td>Reference tree in Newick format containing query genomes placed with pplacer</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).summary.tsv</td>
<td>Classification of query genomes based on their placement in the reference tree, relative evolutionary divergence, and ANI to reference genomes</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).classification_pplacer.tsv</td>
<td>Classification of query genomes based only on their placement in the reference tree</td>
</tr>
<tr>
<td>gtdb.(ar122|bac120).red_dictionary.tsv</td>
<td>Median RED values for taxonomic ranks</td>
</tr>
<tr>
<td>pplacer.(ar122|bac120).json</td>
<td>Output information generated by pplacer in JSON format</td>
</tr>
<tr>
<td>pplacer.(ar122|bac120).out</td>
<td>Output information generated by pplacer</td>
</tr>
</tbody>
</table>
<h4 id="identify">identify<a class="headerlink" href="#identify" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>gtdb.ar122.markers_summary.tsv</td>
<td>Summary of unique, duplicated, and missing markers within the 122 archaeal marker set for each submitted genome</td>
</tr>
<tr>
<td>gtdb.bac120.markers_summary.tsv</td>
<td>Summary of unique, duplicated, and missing markers within the 120 bacterial marker set for each submitted genome</td>
</tr>
<tr>
<td>gtdb.translation_table_summary.tsv</td>
<td>The predicted translation table used for gene calling for each genome</td>
</tr>
<tr>
<td>identify/intermediate_results/marker_genes/</td>
<td>Contains individual genome results for gene calling using Prodigal and gene identification based on TIGRFAM and Pfam HMMs</td>
</tr>
</tbody>
</table>
<h2 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>Required Parameters:
    --bactopia STR          Directory containing Bactopia analysis results for all samples.

    --gtdb STR              Location of a GTDB database. If a database is not found, you must
                                use &#39;--download_gtdb&#39;.

Optional Parameters:
    --include STR           A text file containing sample names to include in the
                                analysis. The expected format is a single sample per line.

    --exclude STR           A text file containing sample names to exclude from the
                                analysis. The expected format is a single sample per line.

    --prefix DIR            Prefix to use for final output files
                                Default: gtdb

    --outdir DIR            Directory to write results to
                                Default: ./

    --max_time INT          The maximum number of minutes a job should run before being halted.
                                Default: 120 minutes

    --max_memory INT        The maximum amount of memory (Gb) allowed to a single process.
                                Default: 32 Gb

    --cpus INT              Number of processors made available to a single
                                process.
                                Default: 1

GTDB-Tk Related Parameters:
    --download_gtdb         Download the latest GTDB database, even it exists.

    --min_perc_aa INT       Filter genomes with an insufficient percentage of AA in the MSA
                                Default: 10

    --recalculate_red       Recalculate RED values based on the reference tree and all added
                                user genomes

    --force_gtdb            Force GTDB to continue processing if an error occurrs on a single
                                genome

    --debug                 Create intermediate files for debugging purposes

Nextflow Related Parameters:
    --condadir DIR          Directory to Nextflow should use for Conda environments
                                Default: Bactopia&#39;s Nextflow directory

    --publish_mode          Set Nextflow&#39;s method for publishing output files. Allowed methods are:
                                &#39;copy&#39; (default)    Copies the output files into the published directory.

                                &#39;copyNoFollow&#39; Copies the output files into the published directory
                                               without following symlinks ie. copies the links themselves.

                                &#39;link&#39;    Creates a hard link in the published directory for each
                                          process output file.

                                &#39;rellink&#39; Creates a relative symbolic link in the published directory
                                          for each process output file.

                                &#39;symlink&#39; Creates an absolute symbolic link in the published directory
                                          for each process output file.

                                Default: copy

    --force                 Nextflow will overwrite existing output files.
                                Default: false

    --conatainerPath        Path to Singularity containers to be used by the &#39;slurm&#39;
                                profile.
                                Default: /opt/bactopia/singularity

    --sleep_time            After reading datases, the amount of time (seconds) Nextflow
                                will wait before execution.
                                Default: 5 seconds

Useful Parameters:
    --version               Print workflow version information
    --help                  Show this message and exit
</code></pre></div>

<h2 id="conda-environment">Conda Environment<a class="headerlink" href="#conda-environment" title="Permanent link">&para;</a></h2>
<p>Below is the command used to create the Conda environment.
<div class="highlight"><pre><span></span><code>conda create -y -n bactopia-gtdb -c conda-forge -c bioconda gtdbtk
</code></pre></div></p>
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p><strong><a href="https://github.com/ParBLiSS/FastANI">FastANI</a></strong><br />
<em>Jain, C., Rodriguez-R, L. M., Phillippy, A. M., Konstantinidis, K. T. &amp; Aluru, S. 
<a href="http://dx.doi.org/10.1038/s41467-018-07641-9">High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries.</a>
 Nat. Commun. 9, 5114 (2018)</em>  </p>
</li>
<li>
<p><strong><a href="http://www.microbesonline.org/fasttree">FastTree 2</a></strong><br />
<em>Price, M. N., Dehal, P. S. &amp; Arkin, A. P. 
<a href="https://dx.doi.org/10.1371%2Fjournal.pone.0009490">FastTree 2 – Approximately Maximum-Likelihood Trees for Large Alignments.</a>
 PLoS One 5, e9490 (2010)</em>  </p>
</li>
<li>
<p><strong><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a></strong><br />
<em>Parks, D. H. et al. 
<a href="https://doi.org/10.1038/nbt.4229">A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life.</a>
 Nat. Biotechnol. 36, 996–1004 (2018) _<br />
_Parks, D. H. et al. 
<a href="https://doi.org/10.1101/771964">Selection of representative genomes for 24,706 bacterial and archaeal species clusters provide a complete genome-based taxonomy.</a>
 bioRxiv 771964 (2019)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk</a></strong><br />
<em>Chaumeil, P.-A., Mussig, A. J., Hugenholtz, P. &amp; Parks, D. H. 
<a href="https://doi.org/10.1093/bioinformatics/btz848">GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database.</a>
 Bioinformatics (2019)</em>  </p>
</li>
<li>
<p><strong><a href="http://hmmer.org/">HMMER3</a></strong><br />
<em>Eddy, S. R. 
<a href="https://doi.org/10.1371/journal.pcbi.1002195">Accelerated Profile HMM Searches.</a> 
PLoS Comput. Biol. 7, e1002195 (2011)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/matsen/pplacer">pplacer</a></strong><br />
<em>Matsen, F. A., Kodner, R. B. &amp; Armbrust, E. V. 
<a href="https://doi.org/10.1186/1471-2105-11-538">pplacer: linear time maximum-likelihood and Bayesian phylogenetic placement of sequences onto a fixed reference tree.</a>
 BMC Bioinformatics 11, 538 (2010)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/hyattpd/Prodigal">Prodigal</a></strong><br />
<em>Hyatt, D. et al. 
<a href="https://doi.org/10.1186/1471-2105-11-119">Prodigal: prokaryotic gene recognition and translation initiation site identification.</a>
 BMC Bioinformatics 11, 119 (2010)</em>  </p>
</li>
</ul>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../fastani/" title="fastani" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                fastani
              </div>
            </div>
          </a>
        
        
          <a href="../ismapper/" title="ismapper" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                ismapper
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright © 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/vendor.92ffa368.min.js"></script>
      <script src="../../assets/javascripts/bundle.5123e3d4.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents"}</script>
      
      <script>
        app = initialize({
          base: "../..",
          features: ["tabs"],
          search: Object.assign({
            worker: "../../assets/javascripts/worker/search.a68abb33.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>