# Acknowledgements

Bactopia is truly a case of *"standing upon the shoulders of giants"*. Nearly 
every component of Bactopia was created by others and made freely available to the public.

I would like to personally extend my many thanks and gratitude to the authors
of these software packages and public datasets. If you've made it this far, I 
owe you a beer üçª (or coffee ‚òï!) if we ever encounter one another in person. 
Really, thank you very much!

!!! info "Please Cite Datasets and Tools"
    If you have used Bactopia in your work, please be sure to cite any datasets or software you may have used. A citation link for each dataset/software has been made available.

    A BibTeX file of each citation is also available at [Bactopia Datasets and Software BibTeX](https://github.com/bactopia/bactopia/blob/master/docs/data/bactopia-datasets-software.bib)

## Public Datasets
Below is a list of public datasets (alphabetical) that could have potentially 
been included during the *[Build Datasets](datasets.md)* step.

### Ariba Reference Datasets
These datasets are available using Ariba's `getref` function. You can learn 
more about this function at [Ariba's Wiki](https://github.com/sanger-pathogens/ariba/wiki/Task:-getref).

* __[ARG-ANNOT](http://en.mediterranee-infection.com/article.php?laref=283%26titre=arg-annot)__  
_Gupta, S. K. et al. [ARG-ANNOT, a new bioinformatic tool to 
discover antibiotic resistance genes in bacterial genomes.](http://www.ncbi.nlm.nih.gov/pubmed/24145532) 
Antimicrob. Agents Chemother. 58, 212‚Äì220 (2014)._  

* __[CARD](https://card.mcmaster.ca/)__  
_Alcock, Brian P., et al. [CARD 2020: antibiotic resistome surveillance with the comprehensive antibiotic resistance database.](https://www.ncbi.nlm.nih.gov/pubmed/31665441) Nucleic acids research 48.D1, D517-D525 (2020)._

* __[MEGARes](https://megares.meglab.org/)__  
_Lakin, S. M. et al. [MEGARes: an antimicrobial resistance database for high 
throughput sequencing](http://www.ncbi.nlm.nih.gov/pubmed/27899569). 
Nucleic Acids Res. 45, D574‚ÄìD580 (2017)._  

* __[NCBI Reference Gene Catalog](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA313047)__  
_Feldgarden, M. et al. [Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates](https://doi.org/10.1128/AAC.00483-19). Antimicrob. Agents Chemother. (2019)_  

* __[plasmidfinder](https://cge.cbs.dtu.dk/services/PlasmidFinder/)__  
_Carattoli, A. et al. [In silico detection and typing of plasmids using 
PlasmidFinder and plasmid multilocus sequence typing.](http://www.ncbi.nlm.nih.gov/pubmed/24777092) 
Antimicrob. Agents Chemother. 58, 3895‚Äì3903 (2014)._

* __[resfinder](https://cge.cbs.dtu.dk//services/ResFinder/)__  
_Zankari, E. et al. [Identification of acquired antimicrobial resistance genes.](http://www.ncbi.nlm.nih.gov/pubmed/22782487) 
J. Antimicrob. Chemother. 67, 2640‚Äì2644 (2012)._  

* __[SRST2](https://github.com/katholt/srst2)__  
_Inouye, M. et al. [SRST2: Rapid genomic surveillance for public health and 
hospital microbiology labs.](http://www.ncbi.nlm.nih.gov/pubmed/25422674) 
Genome Med. 6, 90 (2014)._  

* __[VFDB](http://www.mgc.ac.cn/VFs/)__  
_Chen, L., Zheng, D., Liu, B., Yang, J. & Jin, Q. [VFDB 2016: hierarchical 
and refined dataset for big data analysis--10 years on.](http://www.ncbi.nlm.nih.gov/pubmed/26578559) 
Nucleic Acids Res. 44, D694‚Äì7 (2016)._  

* __[VirulenceFinder](https://cge.cbs.dtu.dk/services/VirulenceFinder/)__  
_Joensen, K. G. et al. [Real-time whole-genome sequencing for routine typing, 
surveillance, and outbreak detection of verotoxigenic Escherichia coli.](http://www.ncbi.nlm.nih.gov/pubmed/24574290) 
J. Clin. Microbiol. 52, 1501‚Äì1510 (2014)._  

### Minmer Datasets
* __[Mash Refseq (release 88) Sketch](https://mash.readthedocs.io/en/latest/data.html)__  
_Ondov, B. D. et al. [Mash Screen: High-throughput sequence containment 
estimation for genome discovery.](https://doi.org/10.1101/557314) bioRxiv 557314 (2019)._  

* __[Sourmash Genbank LCA Signature](https://sourmash.readthedocs.io/en/latest/databases.html)__  
_Titus Brown, C. & Irber, L. [sourmash: a library for MinHash sketching of DNA.](http://joss.theoj.org/papers/10.21105/joss.00027) 
JOSS 1, 27 (2016)._  

### Everything Else
* __[eggNOG 5.0 Database](http://eggnog.embl.de/)__  
_J. Huerta-Cepas, D. Szklarczyk, D. Heller, A. Hern√°ndez-Plaza, S. K. Forslund, H. Cook, D. R. Mende, I. Letunic, T. Rattei, L. J. Jensen, C. von Mering, P. Bork, [eggNOG 5.0: a hierarchical, functionally and phylogenetically annotated orthology resource based on 5090 organisms and 2502 viruses.](https://doi.org/10.1093/nar/gky1085) Nucleic Acids Res. 47, D309‚ÄìD314 (2019)._  

* __[Genome Taxonomy Database](https://gtdb.ecogenomic.org/)__  
_Parks, D. H. et al. 
[A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life.](https://doi.org/10.1038/nbt.4229)
 Nat. Biotechnol. 36, 996‚Äì1004 (2018)_  
_Parks, D. H. et al. 
[Selection of representative genomes for 24,706 bacterial and archaeal species clusters provide a complete genome-based taxonomy.](https://doi.org/10.1101/771964)
 bioRxiv 771964 (2019)_  

* __[NCBI RefSeq Database](https://www.ncbi.nlm.nih.gov/refseq/)__  
_O‚ÄôLeary, N. A. et al. [Reference sequence (RefSeq) database at NCBI: current status, 
taxonomic expansion, and functional annotation](http://dx.doi.org/10.1093/nar/gkv1189). 
Nucleic Acids Res. 44, D733‚Äì45 (2016)._  

* __[PLSDB - A plasmid database](https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/)__  
_Galata, V., Fehlmann, T., Backes, C. & Keller, A. [PLSDB: a resource of complete bacterial 
plasmids](http://dx.doi.org/10.1093/nar/gky1050). 
Nucleic Acids Res. 47, D195‚ÄìD202 (2019)._  

* __[PubMLST.org](https://pubmlst.org/)__  
_Jolley, K. A., Bray, J. E. & Maiden, M. C. J. [Open-access bacterial population genomics: BIGSdb 
software, the PubMLST.org website and their applications](http://dx.doi.org/10.12688/wellcomeopenres.14826.1). 
Wellcome Open Res 3, 124 (2018)._  

* __[SILVA rRNA Database](https://www.arb-silva.de/)__  
_Quast, C. et al. 
[The SILVA ribosomal RNA gene database project: improved data processing and web-based tools.](https://doi.org/10.1093/nar/gks1219) 
Nucleic Acids Res. 41, D590‚Äì6 (2013)_

## Software Included In Bactopia
Below is a list of software (alphabetical) used (directly and indirectly) by Bactopia. A link to the software page as well as the citation (if available) have been included.

* __[AMRFinderPlus](https://github.com/ncbi/amr)__  
Find acquired antimicrobial resistance genes and some point mutations in protein or assembled nucleotide sequences.  
_Feldgarden, M. et al. [Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates](https://doi.org/10.1128/AAC.00483-19). Antimicrob. Agents Chemother. (2019)_  

* __[Aragorn](http://130.235.244.92/ARAGORN/Downloads/)__  
Finds transfer RNA features (tRNA)  
_Laslett D. and B. Canback, [ARAGORN, a program to detect tRNA genes and tmRNA genes in nucleotide sequences.](https://doi.org/10.1093/nar/gkh152) Nucleic Acids Res. 32(1):11-6. (2004)_  

* __[Ariba](https://github.com/sanger-pathogens/ariba)__  
Antimicrobial Resistance Identification By Assembly  
_Hunt, M. et al. [ARIBA: rapid antimicrobial resistance genotyping directly from 
sequencing reads](http://dx.doi.org/10.1099/mgen.0.000131). 
Microb Genom 3, e000131 (2017)._  

* __[ART](https://www.niehs.nih.gov/research/resources/software/biostatistics/art/index.cfm)__  
A set of simulation tools to generate synthetic next-generation sequencing reads  
_W. Huang, L. Li, J. R. Myers, G. T. Marth, [ART: a next-generation sequencing read simulator.](http://dx.doi.org/10.1093/bioinformatics/btr708) Bioinformatics. 28, 593‚Äì594 (2012)._  

* __[assembly-scan](https://github.com/rpetit3/assembly-scan)__  
Generate basic stats for an assembly.  
_Petit III, R. A. [assembly-scan: generate basic stats for an 
assembly](https://github.com/rpetit3/assembly-scan)._  

* __[Barrnap](https://github.com/tseemann/barrnap)__  
Bacterial ribosomal RNA predictor  
_Seemann, T. [Barrnap: Bacterial ribosomal RNA predictor](https://github.com/tseemann/barrnap)_  

* __[BBTools](https://jgi.doe.gov/data-and-tools/bbtools/)__  
BBTools is a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data.  
_Bushnell, B. [BBMap short read aligner, and other bioinformatic tools.](http://sourceforge.net/projects/bbmap/)_  

* __[BCFtools](https://github.com/samtools/bcftools)__  
Utilities for variant calling and manipulating VCFs and BCFs.  
_Danecek, P. et al. [BCFtools - Utilities for variant calling and manipulating VCFs and BCFs.](http://github.com/samtools/bcftools)_  

* __[Bedtools](https://github.com/arq5x/bedtools2)__  
A powerful toolset for genome arithmetic.  
_Quinlan, A. R. & Hall, I. M. [BEDTools: a flexible suite of utilities for 
comparing genomic features](http://dx.doi.org/10.1093/bioinformatics/btq033). 
Bioinformatics 26, 841‚Äì842 (2010)._  

* __[BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi)__  
Basic Local Alignment Search Tool  
_Camacho, C. et al. [BLAST+: architecture and applications](http://dx.doi.org/10.1186/1471-2105-10-421). 
BMC Bioinformatics 10, 421 (2009)._  

* __[Bowtie2](https://github.com/BenLangmead/bowtie2)__  
A fast and sensitive gapped read aligner  
_B. Langmead, S. L. Salzberg, [Fast gapped-read alignment with Bowtie 2.](http://dx.doi.org/10.1038/nmeth.1923) Nat. Methods. 9, 357‚Äì359 (2012)._  

* __[BWA](https://github.com/lh3/bwa/)__  
Burrow-Wheeler Aligner for short-read alignment  
_Li, H. [Aligning sequence reads, clone sequences and assembly contigs with 
BWA-MEM](http://arxiv.org/abs/1303.3997). arXiv [q-bio.GN] (2013)._  

* __[CD-Hit](https://github.com/weizhongli/cdhit)__  
Accelerated for clustering the next-generation sequencing data  
_Li, W. & Godzik, A. [Cd-hit: a fast program for clustering and comparing large sets of protein 
or nucleotide sequences](http://dx.doi.org/10.1093/bioinformatics/btl158). 
Bioinformatics 22, 1658‚Äì1659 (2006)._  
_Fu, L., Niu, B., Zhu, Z., Wu, S. & Li, W. [CD-HIT: accelerated for clustering the next-generation 
sequencing data](http://dx.doi.org/10.1093/bioinformatics/bts565). 
Bioinformatics 28, 3150‚Äì3152 (2012)._  

* __[CheckM](https://github.com/Ecogenomics/CheckM)__  
Assess the quality of microbial genomes recovered from isolates, single cells, and metagenomes  
_D. H. Parks, M. Imelfort, C. T. Skennerton, P. Hugenholtz, G. W. Tyson, [CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes.](http://dx.doi.org/10.1101/gr.186072.114) Genome Res. 25, 1043‚Äì1055 (2015)._  

* __[ClonalFramML](https://github.com/xavierdidelot/ClonalFrameML)__  
Efficient Inference of Recombination in Whole Bacterial Genomes  
_Didelot, X. & Wilson, D. J. 
[ClonalFrameML: efficient inference of recombination in whole bacterial genomes.](http://dx.doi.org/10.1371/journal.pcbi.1004041)
 PLoS Comput. Biol. 11, e1004041 (2015)_  

* __[DIAMOND](https://github.com/bbuchfink/diamond)__  
Accelerated BLAST compatible local sequence aligner.  
_B. Buchfink, C. Xie, D. H. Huson, [Fast and sensitive protein alignment using DIAMOND.](http://dx.doi.org/10.1038/nmeth.3176) Nat. Methods. 12, 59‚Äì60 (2015)._  

* __[eggNOG-mapper](https://github.com/eggnogdb/eggnog-mapper)__  
Fast genome-wide functional annotation through orthology assignment  
_J. Huerta-Cepas, K. Forslund, L. P. Coelho, D. Szklarczyk, L. J. Jensen, C. von Mering, P. Bork, [Fast Genome-Wide Functional Annotation through Orthology Assignment by eggNOG-Mapper.](http://dx.doi.org/10.1093/molbev/msx148) Mol. Biol. Evol. 34, 2115‚Äì2122 (2017)._  

* __[FastANI](https://github.com/ParBLiSS/FastANI)__  
Fast Whole-Genome Similarity (ANI) Estimation  
_Jain, C., Rodriguez-R, L. M., Phillippy, A. M., Konstantinidis, K. T. & Aluru, S. 
[High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries.](http://dx.doi.org/10.1038/s41467-018-07641-9)
 Nat. Commun. 9, 5114 (2018)_  

* __[FastQC](https://github.com/s-andrews/FastQC)__  
A quality control analysis tool for high throughput sequencing data.  
_Andrews, S. [FastQC: a quality control tool for high throughput sequence data.](http://www.bioinformatics.babraham.ac.uk/projects/fastqc)._  

* __[fastq-dl](https://github.com/rpetit3/fastq-dl)__  
Download FASTQ files from SRA or ENA repositories.  
_Petit III, R. A. [fastq-dl: Download FASTQ files from SRA or ENA repositories.](https://github.com/rpetit3/fastq-dl)_  

* __[fastq-scan](https://github.com/rpetit3/fastq-scan)__  
Output FASTQ summary statistics in JSON format  
_Petit III, R. A. [fastq-scan: generate summary statistics of input FASTQ sequences.](https://github.com/rpetit3/fastq-scan)_  

* __[FastTree 2](http://www.microbesonline.org/fasttree)__  
Approximately-maximum-likelihood phylogenetic trees  
_Price, M. N., Dehal, P. S. & Arkin, A. P. 
[FastTree 2 ‚Äì Approximately Maximum-Likelihood Trees for Large Alignments.](https://dx.doi.org/10.1371%2Fjournal.pone.0009490)
 PLoS One 5, e9490 (2010)_  

* __[FLASH](https://ccb.jhu.edu/software/FLASH/)__  
A fast and accurate tool to merge paired-end reads.  
_Magoƒç, T., and S. L. Salzberg, [FLASH: fast length adjustment of short reads to improve genome assemblies.](https://doi.org/10.1093/bioinformatics/btr507) Bioinformatics 27.21 (2011): 2957-2963._  

* __[freebayes](https://github.com/ekg/freebayes)__  
Bayesian haplotype-based genetic polymorphism discovery and genotyping  
_Garrison E., and G. Marth, [Haplotype-based variant detection from short-read sequencing.](https://arxiv.org/abs/1207.3907) arXiv preprint arXiv:1207.3907 [q-bio.GN] (2012)_  

* __[GNU Parallel](https://www.gnu.org/software/parallel/)__  
A shell tool for executing jobs in parallel  
_Tange, O. [GNU Parallel](https://doi.org/10.5281/zenodo.1146014) 2018, March 2018_  

* __[GTDB-Tk](https://github.com/Ecogenomics/GTDBTk)__  
A toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes  
_Chaumeil, P.-A., Mussig, A. J., Hugenholtz, P. & Parks, D. H. 
[GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database.](https://doi.org/10.1093/bioinformatics/btz848)
 Bioinformatics (2019)_  

* __[HMMER](http://hmmer.org/)__  
Biosequence analysis using profile hidden Markov models  
_Eddy, S. R. 
[Accelerated Profile HMM Searches.](https://doi.org/10.1371/journal.pcbi.1002195) 
PLoS Comput. Biol. 7, e1002195 (2011)_  

* __[Infernal](http://eddylab.org/infernal/)__  
Searches DNA sequence databases for RNA structure and sequence similarities  
_Nawrocki, E. P., and S. R. Eddy, [Infernal 1.1: 100-fold faster RNA homology searches.](https://doi.org/10.1093/bioinformatics/btt509) Bioinformatics, 29(22), 2933-2935. (2013)_  

* __[IQ-TREE](https://github.com/Cibiv/IQ-TREE)__  
Efficient phylogenomic software by maximum likelihood  
_L.-T. Nguyen, H.A. Schmidt, A. von Haeseler, B.Q. Minh (2015) 
[IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.](https://doi.org/10.1093/molbev/msu300)
 Mol. Biol. Evol., 32:268-274._  
_S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, L.S. Jermiin (2017) 
[ModelFinder: Fast model selection for accurate phylogenetic estimates.](https://doi.org/10.1038/nmeth.4285) 
Nat. Methods, 14:587-589._  
_D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, L.S. Vinh (2018) [
UFBoot2: Improving the ultrafast bootstrap approximation.](https://doi.org/10.1093/molbev/msx281)
 Mol. Biol. Evol., 35:518‚Äì522._  

* __[ISMapper](https://github.com/jhawkey/IS_mapper)__  
IS mapping software  
_Hawkey, J. et al. [ISMapper: identifying transposase insertion sites in 
bacterial genomes from short read sequence data](http://dx.doi.org/10.1186/s12864-015-1860-2). 
BMC Genomics 16, 667 (2015)._  

* __[Lighter](https://github.com/mourisl/Lighter)__  
Fast and memory-efficient sequencing error corrector  
_Song, L., Florea, L. and B. Langmead, [Lighter: Fast and Memory-efficient Sequencing Error Correction without Counting](http://genomebiology.com/2014/15/11/509/). Genome Biol. 2014 Nov 15;15(11):509._

* __[MAFFT](https://mafft.cbrc.jp/alignment/software/)__  
Multiple alignment program for amino acid or nucleotide sequences  
_Katoh, K. & Standley, D. M. 
[MAFFT multiple sequence alignment software version 7: improvements in performance and usability.](https://doi.org/10.1093/molbev/mst010) 
Mol. Biol. Evol. 30, 772‚Äì780 (2013)_  

* __[Mash](https://github.com/marbl/Mash)__  
Fast genome and metagenome distance estimation using MinHash  
_Ondov, B. D. et al. [Mash: fast genome and metagenome distance 
estimation using MinHash](http://dx.doi.org/10.1186/s13059-016-0997-x). 
Genome Biol. 17, 132 (2016)._  
_Ondov, B. D. et al. [Mash Screen: High-throughput sequence 
containment estimation for genome discovery](http://dx.doi.org/10.1101/557314). 
bioRxiv 557314 (2019)._  

* __[Mashtree](https://github.com/lskatz/mashtree)__  
Create a tree using Mash distances  
_Katz, L. S., Griswold, T., Morrison, S., Caravas, J., Zhang, S., den Bakker, H.C., Deng, X., and Carleton, H. A. [Mashtree: a rapid comparison of whole genome sequence files.](https://doi.org/10.21105/joss.01762) Journal of Open Source Software, 4(44), 1762, (2019)_  

* __[maskrc-svg](https://github.com/kwongj/maskrc-svg)__  
Masks recombination as detected by ClonalFrameML or Gubbins  
_Kwong, J. [maskrc-svg - Masks recombination as detected by ClonalFrameML or Gubbins and draws an SVG.](https://github.com/kwongj/maskrc-svg)_  

* __[McCortex](https://github.com/mcveanlab/mccortex)__  
De novo genome assembly and multisample variant calling  
_Turner, I., Garimella, K. V., Iqbal, Z. and G. McVean, [Integrating long-range 
connectivity information into de Bruijn graphs.](http://dx.doi.org/10.1093/bioinformatics/bty157) 
Bioinformatics 34, 2556‚Äì2565 (2018)._  

* __[MEGAHIT](https://github.com/voutcn/megahit)__  
Ultra-fast and memory-efficient (meta-)genome assembler  
_Li, D., et al. [MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph.](https://doi.org/10.1093/bioinformatics/btv033) Bioinformatics 31.10 (2015): 1674-1676._  

* __[MinCED](https://github.com/ctSkennerton/minced)__  
Mining CRISPRs in Environmental Datasets  
_Skennerton, C. [MinCED: Mining CRISPRs in Environmental Datasets](https://github.com/ctSkennerton/minced)_  

* __[Minimap2](https://github.com/lh3/minimap2)__  
A versatile pairwise aligner for genomic and spliced nucleotide sequences  
_Li, H. [Minimap2: pairwise alignment for nucleotide sequences.](https://doi.org/10.1093/bioinformatics/bty191) Bioinformatics, 34:3094-3100. (2018)_  

* __[ncbi-genome-download](https://github.com/kblin/ncbi-genome-download)__  
Scripts to download genomes from the NCBI FTP servers  
_Blin, K. [ncbi-genome-download: Scripts to download genomes from the NCBI FTP 
servers](https://github.com/kblin/ncbi-genome-download)_  

* __[Nextflow](https://github.com/nextflow-io/nextflow)__  
A DSL for data-driven computational pipelines.  
_Di Tommaso, P., Chatzou, M., Floden, E. W., Barja, P.P., Palumbo, E., Notredame, C., 2017. [Nextflow enables reproducible computational workflows.](https://www.nature.com/articles/nbt.3820.pdf?origin=ppub) Nat. Biotechnol. 35, 316‚Äì319._

* __[nhmmer](http://hmmer.org/)__  
DNA homology search with profile HMMs.  
_Wheeler, T. J. & Eddy, S. R. 
[nhmmer: DNA homology search with profile HMMs.](https://doi.org/10.1093/bioinformatics/btt403)
 Bioinformatics 29, 2487‚Äì2489 (2013)_  

* __[phyloFlash](https://github.com/HRGV/phyloFlash)__  
A pipeline to rapidly reconstruct the SSU rRNAs and explore phylogenetic composition of an illumina (meta)genomic dataset.  
_H. R. Gruber-Vodicka, B.KB. Seah, E. Pruesse. 
[phyloFlash ‚Äî Rapid SSU rRNA profiling and targeted assembly from metagenomes.](https://doi.org/10.1101/521922) 
bioRxiv 521922_  

* __[Pigz](https://zlib.net/pigz/)__  
A parallel implementation of gzip for modern multi-processor, multi-core machines.  
_Adler, M. [pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.](https://zlib.net/pigz/) Jet Propulsion Laboratory (2015)._  

* __[Pilon](https://github.com/broadinstitute/pilon/)__  
An automated genome assembly improvement and variant detection tool  
_Walker, B. J., et al. [Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement.](https://doi.org/10.1371/journal.pone.0112963) PloS one 9.11 (2014): e112963._  

* __[PIRATE](http://github.com/SionBayliss/PIRATE)__  
A toolbox for pangenome analysis and threshold evaluation.  
_S. C. Bayliss, H. A. Thorpe, N. M. Coyle, S. K. Sheppard, E. J. Feil 
[PIRATE: A fast and scalable pangenomics toolbox for clustering diverged orthologues in bacteria.](https://doi.org/10.1093/gigascience/giz119) 
Gigascience. 8 (2019)_  

* __[pplacer](https://github.com/matsen/pplacer)__  
Phylogenetic placement and downstream analysis  
_Matsen, F. A., Kodner, R. B. & Armbrust, E. V. 
[pplacer: linear time maximum-likelihood and Bayesian phylogenetic placement of sequences onto a fixed reference tree.](https://doi.org/10.1186/1471-2105-11-538)
 BMC Bioinformatics 11, 538 (2010)_  

* __[Prodigal](https://github.com/hyattpd/Prodigal)__  
Fast, reliable protein-coding gene prediction for prokaryotic genomes.  
_Hyatt, D., et al. [Prodigal: prokaryotic gene recognition and translation initiation site identification.](https://doi.org/10.1186/1471-2105-11-119) BMC Bioinformatics 11.1 (2010): 119._  

* __[Prokka](https://github.com/tseemann/prokka)__  
Rapid prokaryotic genome annotation  
_Seemann, T. [Prokka: rapid prokaryotic genome annotation](http://dx.doi.org/10.1093/bioinformatics/btu153). 
Bioinformatics 30, 2068‚Äì2069 (2014)._  

* __[QUAST](http://quast.sourceforge.net/)__  
Quality Assessment Tool for Genome  
_A. Gurevich, V. Saveliev, N. Vyahhi, G. Tesler, [QUAST: quality assessment tool for genome assemblies.](http://dx.doi.org/10.1093/bioinformatics/btt086) Bioinformatics. 29, 1072‚Äì1075 (2013)._  

* __[Racon](https://github.com/lbcb-sci/racon)__  
Ultrafast consensus module for raw de novo genome assembly of long uncorrected reads  
_R. Vaser, I. Soviƒá, N. Nagarajan, M. ≈†ikiƒá, [Fast and accurate de novo genome assembly from long uncorrected reads.](http://dx.doi.org/10.1101/gr.214270.116) Genome Res. 27, 737‚Äì746 (2017)._  

* __[RNAmmer](http://www.cbs.dtu.dk/services/RNAmmer/)__  
Consistent and rapid annotation of ribosomal RNA genes  
_Lagesen, K., et al. [RNAmmer: consistent annotation of rRNA genes in genomic sequences.](https://dx.doi.org/10.1093%2Fnar%2Fgkm160) Nucleic Acids Res 35.9: 3100-3108. (2007)_  

* __[Roary](https://github.com/sanger-pathogens/Roary)__  
Rapid large-scale prokaryote pan genome analysis  
_Page, A. J. et al. 
[Roary: rapid large-scale prokaryote pan genome analysis.](https://doi.org/10.1093/bioinformatics/btv421)
 Bioinformatics 31, 3691‚Äì3693 (2015)_  

* __[samclip](https://github.com/tseemann/samclip)__  
Filter SAM file for soft and hard clipped alignments  
_Seemann, T. [Samclip: Filter SAM file for soft and hard clipped alignments](https://github.com/tseemann/samclip)_  

* __[Samtools](https://github.com/samtools/samtools)__  
Tools for manipulating next-generation sequencing data  
_Li, H. et al. [The Sequence Alignment/Map format and SAMtools](http://dx.doi.org/10.1093/bioinformatics/btp352). 
Bioinformatics 25, 2078‚Äì2079 (2009)._

* __[Seqtk](https://github.com/lh3/seqtk)__  
A fast and lightweight tool for processing sequences in the FASTA or FASTQ format.  
_Li, H. [Toolkit for processing sequences in FASTA/Q formats](https://github.com/lh3/seqtk)_  

* __[Shovill](https://github.com/tseemann/shovill)__  
Faster assembly of Illumina reads  
_Seemann, T. [Shovill: De novo assembly pipeline for Illumina paired reads](https://github.com/tseemann/shovill)_  

* __[SignalP](http://www.cbs.dtu.dk/services/SignalP-4.0/)__  
Finds signal peptide features in CDS  
_Petersen, T. N., et al. [SignalP 4.0: discriminating signal peptides from transmembrane regions.](https://doi.org/10.1038/nmeth.1701) Nature methods 8.10: 785.(2011)_  

* __[SKESA](https://github.com/ncbi/SKESA)__  
Strategic Kmer Extension for Scrupulous Assemblies  
_Souvorov, A., Agarwala, R. and D. J. Lipman. [SKESA: strategic k-mer extension for scrupulous assemblies.](https://doi.org/10.1186/s13059-018-1540-z) Genome Biology 19:153 (2018).__  

* __[Snippy](https://github.com/tseemann/snippy)__  
Rapid haploid variant calling and core genome alignment  
_Seemann, T. [Snippy: fast bacterial variant calling from NGS reads](https://github.com/tseemann/snippy)
(2015)_  

* __[SnpEff](http://snpeff.sourceforge.net/)__  
Genomic variant annotations and functional effect prediction toolbox.  
_Cingolani, P., et al. [A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3.](https://doi.org/10.4161/fly.19695) Fly, 6(2), 80-92 (2012)_  

* __[snp-dists](https://github.com/tseemann/snp-dists)__  
Pairwise SNP distance matrix from a FASTA sequence alignment  
_Seemann, T. [snp-dists - Pairwise SNP distance matrix from a FASTA sequence alignment.](https://github.com/tseemann/snp-dists)_  

* __[SNP-sites](https://github.com/sanger-pathogens/snp-sites)__  
Rapidly extracts SNPs from a multi-FASTA alignment.  
_Page, A. J., et al. [SNP-sites: rapid efficient extraction of SNPs from multi-FASTA alignments.](https://dx.doi.org/10.1099%2Fmgen.0.000056) Microbial Genomics 2.4 (2016)._  

* __[Sourmash](https://github.com/dib-lab/sourmash)__  
Compute and compare MinHash signatures for DNA data sets.  
_Titus Brown, C. and L. Irber [sourmash: a library for MinHash sketching 
of DNA](http://dx.doi.org/10.21105/joss.00027). JOSS 1, 27 (2016)._  

* __[SPAdes](https://github.com/ablab/spades)__  
An assembly toolkit containing various assembly pipelines.  
_Bankevich, A., et al. [SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing.](https://doi.org/10.1089/cmb.2012.0021) Journal of computational biology 19.5 (2012): 455-477._  

* __[Trimmomatic](http://www.usadellab.org/cms/index.php?page=trimmomatic)__  
A flexible read trimming tool for Illumina NGS data  
_Bolger, A. M., Lohse, M., and B. Usadel. [Trimmomatic: a flexible trimmer for Illumina sequence data.](https://doi.org/10.1093/bioinformatics/btu170) Bioinformatics 30.15 (2014): 2114-2120._  

* __[Unicycler](https://github.com/rrwick/Unicycler)__  
Hybrid assembly pipeline for bacterial genomes  
_R. R. Wick, L. M. Judd, C. L. Gorrie, K. E. Holt, [Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads.](http://dx.doi.org/10.1371/journal.pcbi.1005595) PLoS Comput. Biol. 13, e1005595 (2017)._  

* __[VCF-Annotator](https://github.com/rpetit3/vcf-annotator)__  
Add biological annotations to variants in a VCF file.  
_Petit III, R. A. [VCF-Annotator: Add biological annotations to variants 
in a VCF file.](https://github.com/rpetit3/vcf-annotator)._  

* __[Vcflib](https://github.com/vcflib/vcflib)__  
a simple C++ library for parsing and manipulating VCF files  
_Garrison, E. [Vcflib: A C++ library for parsing and manipulating VCF files](https://github.com/vcflib/vcflib)_  

* __[Velvet](https://github.com/dzerbino/velvet)__  
Short read de novo assembler using de Bruijn graphs  
_Zerbino, D. R., and E. Birney. [Velvet: algorithms for de novo short read assembly using de Bruijn graphs.](http://www.genome.org/cgi/doi/10.1101/gr.074492.107) Genome research 18.5 (2008): 821-829._  

* __[VSEARCH](https://github.com/torognes/vsearch)__  
Versatile open-source tool for metagenomics  
_Rognes, T., Flouri, T., Nichols, B., Quince, C. & Mah√©, F. 
[VSEARCH: a versatile open source tool for metagenomics.](https://doi.org/10.7717/peerj.2584)
 PeerJ 4, e2584 (2016)_  

* __[vt](https://github.com/atks/vt)__  
A tool set for short variant discovery in genetic sequence data.  
_Tan, A., Abecasis, G. R., and H. M. Kang, [Unified representation of genetic variants.](https://doi.org/10.1093/bioinformatics/btv112) Bioinformatics, 31(13), 2202-2204. (2015)_  



