


<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/workflow-overview/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-5.5.0">
    
    
      
        <title>Workflow Overview - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.b5d04df8.min.css">
      
      
    
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700%7CUbuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
        
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-104199698-2","bactopia.github.io"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}),document.addEventListener("DOMContentSwitch",function(){ga("send","pageview",document.location.pathname)})</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#workflow-overview" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo" aria-label="Bactopia">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            Bactopia
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Workflow Overview
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
        
      
      
        
          

  

<nav class="md-tabs md-tabs--active" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href=".." class="md-tabs__link md-tabs__link--active">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../bactopia-tools/" class="md-tabs__link">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1" checked>
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Bactopia" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../datasets/" title="Build Datasets" class="md-nav__link">
      Build Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Workflow Overview
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 9h14V7H3v2m0 4h14v-2H3v2m0 4h14v-2H3v2m16 0h2v-2h-2v2m0-10v2h2V7h-2m0 6h2v-2h-2v2z"/></svg>
        </span>
      </label>
    
    <a href="./" title="Workflow Overview" class="md-nav__link md-nav__link--active">
      Workflow Overview
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#always-enabled-steps" class="md-nav__link">
    Always Enabled Steps
  </a>
  
    <nav class="md-nav" aria-label="Always Enabled Steps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-fastqs" class="md-nav__link">
    Gather FASTQs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#validate-fastqs" class="md-nav__link">
    Validate FASTQs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#original-summary" class="md-nav__link">
    Original Summary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-size" class="md-nav__link">
    Genome Size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    Quality Control
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-summary" class="md-nav__link">
    QC Summary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#count-31-mers" class="md-nav__link">
    Count 31-mers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-sketch" class="md-nav__link">
    Minmer Sketch
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#de-novo-assembly" class="md-nav__link">
    De novo Assembly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-quality-assessment" class="md-nav__link">
    Assembly Quality Assessment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-annotation" class="md-nav__link">
    Genome Annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#antimicrobial-resistance" class="md-nav__link">
    Antimicrobial Resistance
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dataset-enabled-steps" class="md-nav__link">
    Dataset Enabled Steps
  </a>
  
    <nav class="md-nav" aria-label="Dataset Enabled Steps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#public-datasets" class="md-nav__link">
    Public Datasets
  </a>
  
    <nav class="md-nav" aria-label="Public Datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#call-variants-auto" class="md-nav__link">
    Call Variants (Auto)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#local-assembly" class="md-nav__link">
    Local Assembly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-query" class="md-nav__link">
    Minmer Query
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequence-type" class="md-nav__link">
    Sequence Type
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#user-datasets" class="md-nav__link">
    User Datasets
  </a>
  
    <nav class="md-nav" aria-label="User Datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast-alignment" class="md-nav__link">
    BLAST Alignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#call-variants-user" class="md-nav__link">
    Call Variants (User)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reference-mapping" class="md-nav__link">
    Reference Mapping
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../changelog/" title="Changelog" class="md-nav__link">
      Changelog
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Bactopia Tools" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/eggnog/" title="eggnog" class="md-nav__link">
      eggnog
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/gtdb/" title="gtdb" class="md-nav__link">
      gtdb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/ismapper/" title="ismapper" class="md-nav__link">
      ismapper
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/mashtree/" title="mashtree" class="md-nav__link">
      mashtree
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/pirate/" title="pirate" class="md-nav__link">
      pirate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/roary/" title="roary" class="md-nav__link">
      roary
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#always-enabled-steps" class="md-nav__link">
    Always Enabled Steps
  </a>
  
    <nav class="md-nav" aria-label="Always Enabled Steps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-fastqs" class="md-nav__link">
    Gather FASTQs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#validate-fastqs" class="md-nav__link">
    Validate FASTQs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#original-summary" class="md-nav__link">
    Original Summary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-size" class="md-nav__link">
    Genome Size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    Quality Control
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-summary" class="md-nav__link">
    QC Summary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#count-31-mers" class="md-nav__link">
    Count 31-mers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-sketch" class="md-nav__link">
    Minmer Sketch
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#de-novo-assembly" class="md-nav__link">
    De novo Assembly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-quality-assessment" class="md-nav__link">
    Assembly Quality Assessment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-annotation" class="md-nav__link">
    Genome Annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#antimicrobial-resistance" class="md-nav__link">
    Antimicrobial Resistance
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dataset-enabled-steps" class="md-nav__link">
    Dataset Enabled Steps
  </a>
  
    <nav class="md-nav" aria-label="Dataset Enabled Steps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#public-datasets" class="md-nav__link">
    Public Datasets
  </a>
  
    <nav class="md-nav" aria-label="Public Datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#call-variants-auto" class="md-nav__link">
    Call Variants (Auto)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#local-assembly" class="md-nav__link">
    Local Assembly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-query" class="md-nav__link">
    Minmer Query
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequence-type" class="md-nav__link">
    Sequence Type
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#user-datasets" class="md-nav__link">
    User Datasets
  </a>
  
    <nav class="md-nav" aria-label="User Datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast-alignment" class="md-nav__link">
    BLAST Alignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#call-variants-user" class="md-nav__link">
    Call Variants (User)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reference-mapping" class="md-nav__link">
    Reference Mapping
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/workflow-overview.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  
                
                
                <h1 id="workflow-overview">Workflow Overview<a class="headerlink" href="#workflow-overview" title="Permanent link">&para;</a></h1>
<p>Bactopia is an extensive workflow integrating numerous steps in bacterial genome analysis. Through out the workflow there are steps that are <em>always enabled</em> and <em>dataset enabled</em>. Each of the steps depicted in the image below are described in this section. </p>
<p>A list of software <strong>directly</strong> used in each step is also listed. Please check out the <a href="/acknowledgements/">Acknowledgements</a> section to get the full list of software as well how to download and cite said software.</p>
<p><img alt="Bactopia Workflow" src="../data/bactopia-workflow.png" /></p>
<h2 id="always-enabled-steps">Always Enabled Steps<a class="headerlink" href="#always-enabled-steps" title="Permanent link">&para;</a></h2>
<p>The <em>Always Enabled Steps</em> are always executed by Bactopia. These steps do not depend of external datasets and thus are always enabled.</p>
<h4 id="gather-fastqs">Gather FASTQs<a class="headerlink" href="#gather-fastqs" title="Permanent link">&para;</a></h4>
<p>Specifies exactly where the input FASTQ/FASTAs are coming from. If you are using local inputs (e.g. <code>--R1/--R2</code>, <code>--fastqs</code>) it will verify they can be accessed. </p>
<p>If an accession(s) (<code>--accession</code> or <code>--accessions</code>) was given, the corresponding FASTQs (SRA/ENA) or assemblies (NCBI Assembly) are downloaded in this step. All assemblies will have 2x250bp Illumina reads simulated withour insertions or deletions and a minimum PHRED score of Q33.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/rpetit3/ena-dl">ART</a></td>
<td align="left">Generate simulated Illumina reads from an assembly</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/ena-dl">ena-dl</a></td>
<td align="left">Download FASTQ files from ENA</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/ena-dl">ncbi-genome-download</a></td>
<td align="left">Download GenBank/RefSeq assemblies from NCBI Assembly database</td>
</tr>
</tbody>
</table>
<h4 id="validate-fastqs">Validate FASTQs<a class="headerlink" href="#validate-fastqs" title="Permanent link">&para;</a></h4>
<p>Determines if the FASTQ file contains enough sequencing to continue processing. The <code>--min_reads</code> and <code>--min_basepairs</code> parameters adjust the minimum amount of sequencing required to continue processing. This step does not <em>directly</em> test the validity of the FASTQ format (although, it would fail if the format is invalid!).</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a></td>
<td align="left">Determine total read and basepairs of FASTQ</td>
</tr>
</tbody>
</table>
<h4 id="original-summary">Original Summary<a class="headerlink" href="#original-summary" title="Permanent link">&para;</a></h4>
<p>Produces summary statistics (read lengths, quality scores, etc...) based on the original input FASTQs.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/s-andrews/FastQC">FastQC</a></td>
<td align="left">Generates a HTML report of original FASTQ summary statistics</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a></td>
<td align="left">Generates original FASTQ summary statistics in JSON format</td>
</tr>
</tbody>
</table>
<h4 id="genome-size">Genome Size<a class="headerlink" href="#genome-size" title="Permanent link">&para;</a></h4>
<p>The genome size is by various programs in the Bactopia workflow. By default, if no genome size is given one is estimated using Mash. Otherwise, a specific genome size can be specified or completely disabled using the <code>--genome_size</code> parameter. See <a href="/usage-basic/#-genome_size">Genome Size Parameter</a> to learn more about specifying the genome size. </p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/marbl/Mash">Mash</a></td>
<td align="left">If not given, estimates genome size of sample</td>
</tr>
</tbody>
</table>
<h4 id="quality-control">Quality Control<a class="headerlink" href="#quality-control" title="Permanent link">&para;</a></h4>
<p>The input FASTQs go through a few clean up steps. First, Illumina related adapters and phiX contaminants are removed. Then reads that fail to pass length and/or quality requirements are filtered out. If the genome size is available, sequence error-corrections are made and the total sequencing is reduced to a specified coverage. After this step, all downstream analyses are based on the QC'd FASTQ and the original is no longer used.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://jgi.doe.gov/data-and-tools/bbtools/">BBTools</a></td>
<td align="left">Removes Illumina adapters, phiX contaminants, filters reads based on length and quality score, and reduces inputs to a specified coverage.</td>
</tr>
<tr>
<td><a href="https://github.com/mourisl/Lighter">Lighter</a></td>
<td align="left">Corrects sequencing errors</td>
</tr>
</tbody>
</table>
<h4 id="qc-summary">QC Summary<a class="headerlink" href="#qc-summary" title="Permanent link">&para;</a></h4>
<p>Produces summary statistics (read lengths, quality scores, etc...) based on the final set of QC'd FASTQs.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/s-andrews/FastQC">FastQC</a></td>
<td align="left">Generates a HTML report of QC'd FASTQ summary statistics</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a></td>
<td align="left">Generates QC'd FASTQ summary statistics in JSON format</td>
</tr>
</tbody>
</table>
<h4 id="count-31-mers">Count 31-mers<a class="headerlink" href="#count-31-mers" title="Permanent link">&para;</a></h4>
<p>All 31 basepair (31-mers) sequences are counted and the singletons (those 31-mers only counted once) are filtered out.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/mcveanlab/mccortex">McCortex</a></td>
<td align="left">Counts 31-mers in the input FASTQ</td>
</tr>
</tbody>
</table>
<h4 id="minmer-sketch">Minmer Sketch<a class="headerlink" href="#minmer-sketch" title="Permanent link">&para;</a></h4>
<p>A minmer sketch and signature is created based on the QC'd FASTQs for multiple values of <em>k</em>. If datasets are available, the sketches/signatures are used for further downstream analysis.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/marbl/Mash">Mash</a></td>
<td align="left">Produces a sketch (<em>k</em>=21,31) of tje QC'd FASTQ</td>
</tr>
<tr>
<td><a href="https://github.com/dib-lab/sourmash">Sourmash</a></td>
<td align="left">Produces a signature (<em>k</em>=21,31,51) of the QC'd FASTQ</td>
</tr>
</tbody>
</table>
<h4 id="de-novo-assembly">De novo Assembly<a class="headerlink" href="#de-novo-assembly" title="Permanent link">&para;</a></h4>
<p>The QC'd FASTQs are assembled using the <a href="https://github.com/tseemann/shovill">Shovill</a> pipeline. This allows for a seamless assembly process using <a href="https://github.com/voutcn/megahit">MEGAHIT</a>, <a href="https://github.com/ncbi/SKESA">SKESA</a>, <a href="https://github.com/ablab/spades">SPAdes</a> or <a href="https://github.com/dzerbino/velvet">Velvet</a>. Alternatively, if long reads are available to complement Illumina paired-end reads, hybrid assembly is available through <a href="https://github.com/rrwick/Unicycler">Unicycler</a>.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/rpetit3/assembly-scan">assembly-scan</a></td>
<td align="left">Generates summary statistics of the final assembly</td>
</tr>
<tr>
<td><a href="https://github.com/tseemann/shovill">Shovill</a></td>
<td align="left">Manages multiple steps in the Illumina assembly process</td>
</tr>
<tr>
<td><a href="https://github.com/rrwick/Unicycler">Unicycler</a></td>
<td align="left">Manages multiple steps in the hybrid assembly process</td>
</tr>
</tbody>
</table>
<h4 id="assembly-quality-assessment">Assembly Quality Assessment<a class="headerlink" href="#assembly-quality-assessment" title="Permanent link">&para;</a></h4>
<p>After assembly, the <em>de novo</em> assembly is assessed for its biological (e.g. containment &amp; contamination) as well as its technical (e.g. misassemblies and errors) quality using <a href="https://github.com/Ecogenomics/CheckM">CheckM</a> and <a href="http://quast.sourceforge.net/">QUAST</a>.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/Ecogenomics/CheckM">CheckM</a></td>
<td align="left">Assess the biological quality of a <em>de novo</em> assembly based on presence of marker genes</td>
</tr>
<tr>
<td><a href="http://quast.sourceforge.net/">QUAST</a></td>
<td align="left">Gives a summary on the technical (e.g. misassemblies etc) quality of a <em>de novo</em> assembly</td>
</tr>
</tbody>
</table>
<h4 id="genome-annotation">Genome Annotation<a class="headerlink" href="#genome-annotation" title="Permanent link">&para;</a></h4>
<p>Genes are predicted and annotated from the assembled genome using <a href="https://github.com/tseemann/prokka">Prokka</a>. If available, a <a href="/datasets/#species-specific">clustered RefSeq protein set</a> is used for the first pass of annotation.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/tseemann/prokka">Prokka</a></td>
<td align="left">Predicts and annotates assembled genomes</td>
</tr>
</tbody>
</table>
<h4 id="antimicrobial-resistance">Antimicrobial Resistance<a class="headerlink" href="#antimicrobial-resistance" title="Permanent link">&para;</a></h4>
<p>Searches for antimicrobial resistance genes and assosiated point mutations in the annotated gene and protein sequences. If datasets are available, <a href="/workflow-overview/#local-assembly">local assemblies</a> can also be used to predict antibiotic resistance.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/ncbi/amr">AMRFinderPlus</a></td>
<td align="left">Predicts antimicrobial resistance based on genes and point mutations</td>
</tr>
</tbody>
</table>
<h2 id="dataset-enabled-steps">Dataset Enabled Steps<a class="headerlink" href="#dataset-enabled-steps" title="Permanent link">&para;</a></h2>
<p>The remaining <em>Dataset Enabled Steps</em> require supplemental datasets to be available to be executed. There are many datasets available that Bactopia can take advantage of. To learn more about setting up these datasets, check out <a href="/datasets/#build-datasets">Build Datasets</a>. These datasets can be broken into two groups, <em>Public Datasets</em> and <em>User Datasets</em>.</p>
<h3 id="public-datasets">Public Datasets<a class="headerlink" href="#public-datasets" title="Permanent link">&para;</a></h3>
<p><a href="/datasets/#general">Publicly available datasets</a> can be used for further analysis.</p>
<h4 id="call-variants-auto">Call Variants (Auto)<a class="headerlink" href="#call-variants-auto" title="Permanent link">&para;</a></h4>
<p>Variants are predicted using <a href="https://github.com/tseemann/snippy">Snippy</a>. The QC'd FASTQs are aligned to the nearest (based on Mash distance) RefSeq completed genome. By default, only the nearest genome is selected, but multiple genomes can be selected (<code>--max_references</code>) or this feature can be completely disabled (<code>disable_auto_variants</code>).</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/arq5x/bedtools2">Bedtools</a></td>
<td align="left">Generates the per-base coverage of the reference alignment</td>
</tr>
<tr>
<td><a href="https://github.com/kblin/ncbi-genome-download">NCBI Genome Download</a></td>
<td align="left">Downloads the RefSeq completed genome</td>
</tr>
<tr>
<td><a href="https://github.com/tseemann/snippy">Snippy</a></td>
<td align="left">Manages multiple steps in the haploid variant calling process</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/vcf-annotator">vcf-annotator</a></td>
<td align="left">Adds annotations from reference GenBank to the final VCF</td>
</tr>
</tbody>
</table>
<h4 id="local-assembly">Local Assembly<a class="headerlink" href="#local-assembly" title="Permanent link">&para;</a></h4>
<p>Using available <a href="/acknowledgements/#ariba-reference-datasets">Ariba reference datasets</a>, determines which reference sequences were found with an additional detailed report summarizing the results.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/sanger-pathogens/ariba">Ariba</a></td>
<td align="left">Creates local assemblies of reference sequences</td>
</tr>
</tbody>
</table>
<h4 id="minmer-query">Minmer Query<a class="headerlink" href="#minmer-query" title="Permanent link">&para;</a></h4>
<p>Screens QC'd FASTQs and signatures against available <a href="/acknowledgements/#minmer-datasets">Minmer Datasets</a>.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/marbl/Mash">Mash</a></td>
<td align="left">Screens against RefSeq and/or PLSDB sketches</td>
</tr>
<tr>
<td><a href="https://github.com/dib-lab/sourmash">Sourmash</a></td>
<td align="left">Screens signature against GenBank</td>
</tr>
</tbody>
</table>
<h4 id="sequence-type">Sequence Type<a class="headerlink" href="#sequence-type" title="Permanent link">&para;</a></h4>
<p>Uses a <a href="https://pubmlst.org/">PubMLST.org</a> MLST schema to determine the sequence type of the sample.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/sanger-pathogens/ariba">Ariba</a></td>
<td align="left">Runs QC'd FASTQ against a MLST database</td>
</tr>
<tr>
<td><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></td>
<td align="left">Aligns MLST loci against the assembled genome</td>
</tr>
</tbody>
</table>
<h3 id="user-datasets">User Datasets<a class="headerlink" href="#user-datasets" title="Permanent link">&para;</a></h3>
<p>Another option is for users to <a href="/datasets/#user-populated-folders">provide their own data</a> to include in the analysis.</p>
<h4 id="blast-alignment">BLAST Alignment<a class="headerlink" href="#blast-alignment" title="Permanent link">&para;</a></h4>
<p>Each gene, protein, or primer sequence provided by the user is aligned against the assembled genome.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></td>
<td align="left">Aligns reference sequences against the assembled genome</td>
</tr>
</tbody>
</table>
<h4 id="call-variants-user">Call Variants (User)<a class="headerlink" href="#call-variants-user" title="Permanent link">&para;</a></h4>
<p>Uses the same procedure as <a href="/workflow-overview/#call-variants-auto">Call Variants (Auto)</a>, except variants are called against each reference provided by the user.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/arq5x/bedtools2">Bedtools</a></td>
<td align="left">Generates the per-base coverage of the reference alignment</td>
</tr>
<tr>
<td><a href="https://github.com/tseemann/snippy">Snippy</a></td>
<td align="left">Manages multiple steps in the haploid variant calling process</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/vcf-annotator">vcf-annotator</a></td>
<td align="left">Adds annotations from reference GenBank to the final VCF</td>
</tr>
</tbody>
</table>
<h4 id="reference-mapping">Reference Mapping<a class="headerlink" href="#reference-mapping" title="Permanent link">&para;</a></h4>
<p>Aligns the QC'd FASTQs to each sequence provided by the user.</p>
<table>
<thead>
<tr>
<th>Software</th>
<th align="left">Usage</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/arq5x/bedtools2">Bedtools</a></td>
<td align="left">Generates the per-base coverage of the reference alignment</td>
</tr>
<tr>
<td><a href="https://github.com/lh3/bwa/">BWA</a></td>
<td align="left">Aligns QC'd FASTQ to a reference sequence</td>
</tr>
<tr>
<td><a href="https://github.com/samtools/samtools">Samtools</a></td>
<td align="left">Converts alignment from SAM to BAM</td>
</tr>
</tbody>
</table>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../usage-basic/" title="Basic Usage" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Basic Usage
              </div>
            </div>
          </a>
        
        
          <a href="../output-overview/" title="Output Overview" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Output Overview
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright © 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/vendor.92ffa368.min.js"></script>
      <script src="../assets/javascripts/bundle.5123e3d4.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents"}</script>
      
      <script>
        app = initialize({
          base: "..",
          features: ["tabs"],
          search: Object.assign({
            worker: "../assets/javascripts/worker/search.a68abb33.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>