


<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/bactopia-tools/roary/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-5.5.0">
    
    
      
        <title>roary - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.b5d04df8.min.css">
      
      
    
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700%7CUbuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
        
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-104199698-2","bactopia.github.io"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}),document.addEventListener("DOMContentSwitch",function(){ga("send","pageview",document.location.pathname)})</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bactopia-tools-roary" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo" aria-label="Bactopia">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            Bactopia
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              roary
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
        
      
      
        
          

  

<nav class="md-tabs md-tabs--active" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../.." class="md-tabs__link">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../" class="md-tabs__link md-tabs__link--active">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Bactopia" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../.." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../datasets/" title="Build Datasets" class="md-nav__link">
      Build Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../workflow-overview/" title="Workflow Overview" class="md-nav__link">
      Workflow Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Bactopia Tools" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../eggnog/" title="eggnog" class="md-nav__link">
      eggnog
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../gtdb/" title="gtdb" class="md-nav__link">
      gtdb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../ismapper/" title="ismapper" class="md-nav__link">
      ismapper
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mashtree/" title="mashtree" class="md-nav__link">
      mashtree
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pirate/" title="pirate" class="md-nav__link">
      pirate
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        roary
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 9h14V7H3v2m0 4h14v-2H3v2m0 4h14v-2H3v2m16 0h2v-2h-2v2m0-10v2h2V7h-2m0 6h2v-2h-2v2z"/></svg>
        </span>
      </label>
    
    <a href="./" title="roary" class="md-nav__link md-nav__link--active">
      roary
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav" aria-label="Output Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav" aria-label="Directory Description">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clonalframe" class="md-nav__link">
    clonalframe
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#iqtree" class="md-nav__link">
    iqtree
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#refseq" class="md-nav__link">
    refseq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#roary" class="md-nav__link">
    roary
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conda-environment" class="md-nav__link">
    Conda Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav" aria-label="Output Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav" aria-label="Directory Description">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clonalframe" class="md-nav__link">
    clonalframe
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#iqtree" class="md-nav__link">
    iqtree
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#refseq" class="md-nav__link">
    refseq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#roary" class="md-nav__link">
    roary
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conda-environment" class="md-nav__link">
    Conda Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/bactopia-tools/roary.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  
                
                
                <h1 id="bactopia-tools-roary">Bactopia Tools - <em>roary</em><a class="headerlink" href="#bactopia-tools-roary" title="Permanent link">&para;</a></h1>
<p>The <code>roary</code> tool allows you to create a pan-genome of your samples using 
<a href="https://github.com/sanger-pathogens/Roary/">Roary</a>. </p>
<p>Often times, you may
also want to include completed genomes in your pan-genome analysis. This 
is possible with the <code>roary</code> tool. If you use the <code>--species</code> parameter,
all completed genomes available from RefSeq will be downloaded with 
<a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a> and 
reannotated with <a href="https://github.com/tseemann/prokka">Prokka</a> (to make 
compatible gffs). </p>
<p>You can also use the core genome alignment, produced by Roary, to create a 
core genome phylogeny using <a href="https://github.com/xavierdidelot/ClonalFrameML">ClonalFrameMl</a>, 
<a href="https://github.com/kwongj/maskrc-svg">maskrc-svg</a>, and <a href="http://www.iqtree.org/">IQ-TREE</a>.
The core genome pair-wise SNP distance for each sample is also calculated with
<a href="https://github.com/tseemann/snp-dists">snp-dists</a>.</p>
<h2 id="example">Example<a class="headerlink" href="#example" title="Permanent link">&para;</a></h2>
<p>The following command will run Roary, on a set of samples in the <em>include</em> file. Then it 
will create a phylogenetic tree based on the core-genome alignment.
<div class="highlight"><pre><span></span><code>bactopia tools roary \
    --bactopia ~/bactopia-tutorial/bactopia \
    --include ~/bactopia-tutorial/GCF_900475245-include.txt \
    --cpus 4 \
    --n
</code></pre></div></p>
<h2 id="output-overview">Output Overview<a class="headerlink" href="#output-overview" title="Permanent link">&para;</a></h2>
<p>Below is the default output structure for the <code>roary</code> tool. Where possible the 
file descriptions below were modified from a tools description.
<div class="highlight"><pre><span></span><code>bactopia-tools
└──roary/
   └── ${PREFIX}
       ├── bactopia-info
       │   ├── roary-report.html
       │   ├── roary-timeline.html
       │   └── roary-trace.txt
       ├── clonalframe
       │   ├── clonalframe.emsim.txt
       │   ├── clonalframe.em.txt
       │   ├── clonalframe.importation_status.txt
       │   ├── clonalframe.labelled_tree.newick
       │   ├── clonalframe.ML_sequence.fasta
       │   ├── clonalframe.position_cross_reference.txt
       │   ├── core_gene_alignment-masked.aln.gz
       │   ├── start-tree.bionj
       │   ├── start-tree.ckp.gz
       │   ├── start-tree.iqtree
       │   ├── start-tree.log
       │   ├── start-tree.mldist
       │   ├── start-tree.model.gz
       │   └── start-tree.treefile
       ├── core-genome.aligned.fa.gz
       ├── core-genome.distance.txt
       ├── core-genome.iqtree
       ├── iqtree
       │   ├── core-genome.alninfo
       │   ├── core-genome.bionj
       │   ├── core-genome.ckp.gz
       │   ├── core-genome.contree
       │   ├── core-genome.iqtree
       │   ├── core-genome.log
       │   ├── core-genome.mldist
       │   ├── core-genome.model.gz
       │   ├── core-genome.splits.nex
       │   ├── core-genome.treefile
       │   └── core-genome.ufboot
       ├── refseq
       │   ├── fasta
       │   │   └── GCF_900475245.fna
       │   └── gff
       │       └── GCF_900475245.gff
       └── roary
           ├── accessory_binary_genes.fa
           ├── accessory_binary_genes.fa.newick
           ├── accessory_graph.dot
           ├── accessory.header.embl
           ├── accessory.tab
           ├── blast_identity_frequency.Rtab
           ├── clustered_proteins
           ├── conserved_vs_total_genes.png
           ├── core_accessory_graph.dot
           ├── core_accessory.header.embl
           ├── core_accessory.tab
           ├── core_alignment_header.embl
           ├── core_gene_alignment.aln.gz
           ├── gene_presence_absence.csv
           ├── gene_presence_absence.Rtab
           ├── number_of_conserved_genes.Rtab
           ├── number_of_genes_in_pan_genome.Rtab
           ├── number_of_new_genes.Rtab
           ├── number_of_unique_genes.Rtab
           ├── pan_genome_reference.fa
           ├── Rplots.pdf
           ├── summary_statistics.txt
           └── unique_vs_new_genes.png
</code></pre></div></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>core-genome.aligned.fa.gz</td>
<td>A multiple sequence alignment FASTA of the core genome</td>
</tr>
<tr>
<td>core-genome.distance.txt</td>
<td>Core genome Pair-wise SNP distance for each sample</td>
</tr>
<tr>
<td>core-genome.iqtree</td>
<td>Full result of the IQ-TREE core genome phylogeny</td>
</tr>
</tbody>
</table>
<h3 id="directory-description">Directory Description<a class="headerlink" href="#directory-description" title="Permanent link">&para;</a></h3>
<h4 id="bactopia-info">bactopia-info<a class="headerlink" href="#bactopia-info" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>roary-report.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#execution-report">Execution Report</a></td>
</tr>
<tr>
<td>roary-timeline.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#timeline-report">Timeline Report</a></td>
</tr>
<tr>
<td>roary-trace.txt</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">Trace</a> report</td>
</tr>
</tbody>
</table>
<h4 id="clonalframe">clonalframe<a class="headerlink" href="#clonalframe" title="Permanent link">&para;</a></h4>
<p>Where possible descriptions were taken from the <a href="https://github.com/xavierdidelot/clonalframeml/wiki">ClonalFrameML Wiki</a>, 
IQ-TREE's <a href="https://github.com/Cibiv/IQ-TREE/wiki/Command-Reference">Command Reference</a>
page, <a href="https://github.com/Cibiv/IQ-TREE/wiki/Web-Server-Tutorial">Web Server Tutorial</a> page, 
and the <a href="http://www.iqtree.org/doc/Tutorial">Tutorial</a> page.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>clonalframe.emsim.txt</td>
<td>The bootstrapped values for the three parameters R/theta, nu and delta</td>
</tr>
<tr>
<td>clonalframe.em.txt</td>
<td>The point estimates for R/theta, nu, delta and the branch lengths</td>
</tr>
<tr>
<td>clonalframe.importation_status.txt</td>
<td>The list of reconstructed recombination events</td>
</tr>
<tr>
<td>clonalframe.labelled_tree.newick</td>
<td>The output tree with all nodes labelled so that they can be referred to in other files</td>
</tr>
<tr>
<td>clonalframe.ML_sequence.fasta</td>
<td>The sequence reconstructed by maximum likelihood for all internal nodes of the phylogeny, as well as for all missing data in the input sequences</td>
</tr>
<tr>
<td>clonalframe.position_cross_reference.txt</td>
<td>A vector of comma-separated values indicating for each location in the input sequence file the corresponding position in the sequences in the output <em>ML_sequence.fasta</em> file</td>
</tr>
<tr>
<td>core_gene_alignment-masked.aln.gz</td>
<td>A core-genome alignment with the recomination masked</td>
</tr>
<tr>
<td>start-tree.bionj</td>
<td>A neighbor joining tree produced by BIONJ</td>
</tr>
<tr>
<td>start-tree.ckp.gz</td>
<td>IQ-TREE writes a checkpoint file</td>
</tr>
<tr>
<td>start-tree.iqtree</td>
<td>Full result of the run, this is the main report file</td>
</tr>
<tr>
<td>start-tree.log</td>
<td>Run log</td>
</tr>
<tr>
<td>start-tree.mldist</td>
<td>Contains the likelihood distances</td>
</tr>
<tr>
<td>start-tree.model.gz</td>
<td>Information about all models tested</td>
</tr>
<tr>
<td>start-tree.treefile</td>
<td>Maximum likelihood tree in NEWICK format, can be visualized with treeviewer programs</td>
</tr>
</tbody>
</table>
<h4 id="iqtree">iqtree<a class="headerlink" href="#iqtree" title="Permanent link">&para;</a></h4>
<p>Where possible descriptions were taken from 
IQ-TREE's <a href="https://github.com/Cibiv/IQ-TREE/wiki/Command-Reference">Command Reference</a>
page, <a href="https://github.com/Cibiv/IQ-TREE/wiki/Web-Server-Tutorial">Web Server Tutorial</a> page, 
and the <a href="http://www.iqtree.org/doc/Tutorial">Tutorial</a> page.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>core-genome.alninfo</td>
<td>Alignment site statistics</td>
</tr>
<tr>
<td>core-genome.bionj</td>
<td>A neighbor joining tree produced by BIONJ</td>
</tr>
<tr>
<td>core-genome.ckp.gz</td>
<td>IQ-TREE writes a checkpoint file</td>
</tr>
<tr>
<td>core-genome.contree</td>
<td>Consensus tree with assigned branch supports where branch lengths are optimized on the original alignment; printed if Ultrafast Bootstrap is selected</td>
</tr>
<tr>
<td>core-genome.iqtree</td>
<td>Full result of the run, this is the main report file</td>
</tr>
<tr>
<td>core-genome.log</td>
<td>Run log</td>
</tr>
<tr>
<td>core-genome.mldist</td>
<td>Contains the likelihood distances</td>
</tr>
<tr>
<td>core-genome.model.gz</td>
<td>Information about all models tested</td>
</tr>
<tr>
<td>core-genome.splits.nex</td>
<td>Support values in percentage for all splits (bipartitions), computed as the occurence frequencies in the bootstrap trees</td>
</tr>
<tr>
<td>core-genome.treefile</td>
<td>Maximum likelihood tree in NEWICK format, can be visualized with treeviewer programs</td>
</tr>
<tr>
<td>core-genome.ufboot</td>
<td>Trees created during the bootstrap steps</td>
</tr>
</tbody>
</table>
<h4 id="refseq">refseq<a class="headerlink" href="#refseq" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.fna</td>
<td>FASTA formated genome downloaded from NCBI Assembly database.</td>
</tr>
<tr>
<td>.gff</td>
<td>GFF output from the Prokka re-annotation of the reference FASTA</td>
</tr>
</tbody>
</table>
<h4 id="roary">roary<a class="headerlink" href="#roary" title="Permanent link">&para;</a></h4>
<p>Where possible descriptions were taken from the <a href="http://sanger-pathogens.github.io/Roary/">Roary Documentation</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>accessory_binary_genes.fa</td>
<td>A FASTA file with binary presence and absence of accessory genes</td>
</tr>
<tr>
<td>accessory_binary_genes.fa.newick</td>
<td>A tree created using the binary presence and absence of accessory genes</td>
</tr>
<tr>
<td>accessory_graph.dot</td>
<td>A graph in DOT format of how genes are linked together at the contig level in the accessory genome</td>
</tr>
<tr>
<td>accessory.header.embl</td>
<td>Tab/EMBL formatted file of accessory genes</td>
</tr>
<tr>
<td>accessory.tab</td>
<td>Tab/EMBL formatted file of accessory genes</td>
</tr>
<tr>
<td>blast_identity_frequency.Rtab</td>
<td>Blast results for percentage idenity graph</td>
</tr>
<tr>
<td>clustered_proteins</td>
<td>Groups file where each line lists the sequences in a cluster</td>
</tr>
<tr>
<td>conserved_vs_total_genes.png</td>
<td>Plot compairing conserved genes and total genes</td>
</tr>
<tr>
<td>core_accessory_graph.dot</td>
<td>A graph in DOT format of how genes are linked together at the contig level in the pan genome</td>
</tr>
<tr>
<td>core_accessory.header.embl</td>
<td>Tab/EMBL formatted file of core genes</td>
</tr>
<tr>
<td>core_accessory.tab</td>
<td>Tab/EMBL formatted file of core genes</td>
</tr>
<tr>
<td>core_alignment_header.embl</td>
<td>Tab/EMBL formatted file of core genome alignment</td>
</tr>
<tr>
<td>core_gene_alignment.aln.gz</td>
<td>A multi-FASTA alignment of all of the core genes</td>
</tr>
<tr>
<td>gene_presence_absence.csv</td>
<td>Lists each gene and which samples it is present in</td>
</tr>
<tr>
<td>gene_presence_absence.Rtab</td>
<td>Tab delimited binary matrix with the presence and absence of each gene in each sample</td>
</tr>
<tr>
<td>number_of_conserved_genes.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>number_of_genes_in_pan_genome.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>number_of_new_genes.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>number_of_unique_genes.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>pan_genome_reference.fa</td>
<td>FASTA file which contains a single representative nucleotide sequence from each of the clusters in the pan genome (core and accessory)</td>
</tr>
<tr>
<td>Rplots.pdf</td>
<td>PDF containing each plot</td>
</tr>
<tr>
<td>summary_statistics.txt</td>
<td>Number of genes in the core and accessory</td>
</tr>
<tr>
<td>unique_vs_new_genes.png</td>
<td>Plot compairing new vs old genes</td>
</tr>
</tbody>
</table>
<h2 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>Required Parameters:
    --bactopia STR          Directory containing Bactopia analysis results for all samples.

Optional Parameters:
    --include STR           A text file containing sample names to include in the
                                analysis. The expected format is a single sample per line.

    --exclude STR           A text file containing sample names to exclude from the
                                analysis. The expected format is a single sample per line.

    --prefix DIR            Prefix to use for final output files
                                Default: core-genome

    --outdir DIR            Directory to write results to
                                Default: ./

    --max_time INT          The maximum number of minutes a job should run before being halted.
                                Default: 2880 minutes

    --max_memory INT        The maximum amount of memory (Gb) allowed to a single process.
                                Default: 64 Gb

    --cpus INT              Number of processors made available to a single
                                process.
                                Default: 1

RefSeq Assemblies Related Parameters:
    --assembly              A single assembly, or directory of assemblies to be included in the
                                pan-genome analysis. If compressed, gzip and the &quot;.gz&quot; extension
                                must be used.

    --assembly_pattern      If a directory is given, use the given pattern to match assemblies.
                                Default: *.fna

    --species STR           The name of the species to download RefSeq assemblies for. This
                                is a completely optional step and is meant to supplement
                                your dataset with high-quality completed genomes.

    --accession STR         A NCBI Assembly database RefSeq accession to be downloaded and included
                                in the pan-genome analysis.

    --limit INT             Limit the number of RefSeq assemblies to download. If the the
                                number of available genomes exceeds the given limit, a 
                                random subset will be selected.
                                Default: Download all available genomes

    --only_completed        Pan-genome will be created using only the completed RefSeq genomes.    

    --prokka_evalue STR     Similarity e-value cut-off
                                Default: 1e-09

    --prokka_coverage INT   Minimum coverage on query protein
                                 Default: 80

Roary Related Parameters:
    --o STR                 Clusters output filename
                                Default: clustered_proteins

    --n                     Execute a fast core gene alignment with MAFFT
                                Default: Use PRANK

    --i INT                 Minimum percentage identity for blastp
                                Default: 95

    --cd INT                Percentage of isolates a gene must be in to be core
                                Default: 99%

    --g INT                 Maximum number of clusters
                                Default: 50000

    --s                     Do not split paralogs
                                Default: false

    --ap                    Allow paralogs in core alignment
                                Default: false

    --iv STR                Change the MCL inflation value
                                Default: 1.5

IQ-TREE Related Parameters:
    --skip_phylogeny        Skip the creation a core-genome based phylogeny

    --m STR                 Substitution model name
                                Default: MFP

    --bb INT                Ultrafast bootstrap replicates
                                Default: 1000

    --alrt INT              SH-like approximate likelihood ratio test replicates
                                Default: 1000

    --asr                   Ancestral state reconstruction by empirical Bayes
                                Default: false

    --iqtree_opts STR       Extra IQ-TREE options in quotes.
                                Default: &#39;&#39;

ClonalFrameML Related Parameters:
    --skip_clonalframe      Skip the ClonalFrameML and use the original core-genome
                                alignment for the final tree.

    --emsim INT             Number of simulations to estimate uncertainty in the EM results.
                                Default: 100

    --clonal_opts STR       Extra ClonalFrameML options in quotes.
                                Default: &#39;&#39;

SNP-Dists Related Parameters:
    --a                     Count all differences not just [AGTC]
                                Default: false

    --b                     Blank top left corner cell
                                Default: false

    --c                     Output CSV instead of TSV
                                Default: false

    --k                     Keep case, don&#39;t uppercase all letters
                                Default: false

Nextflow Related Parameters:
    --condadir DIR          Directory to Nextflow should use for Conda environments
                                Default: Bactopia&#39;s Nextflow directory

    --publish_mode          Set Nextflow&#39;s method for publishing output files. Allowed methods are:
                                &#39;copy&#39; (default)    Copies the output files into the published directory.

                                &#39;copyNoFollow&#39; Copies the output files into the published directory
                                               without following symlinks ie. copies the links themselves.

                                &#39;link&#39;    Creates a hard link in the published directory for each
                                          process output file.

                                &#39;rellink&#39; Creates a relative symbolic link in the published directory
                                          for each process output file.

                                &#39;symlink&#39; Creates an absolute symbolic link in the published directory
                                          for each process output file.

                                Default: copy

    --force                 Nextflow will overwrite existing output files.
                                Default: null

    --conatainerPath        Path to Singularity containers to be used by the &#39;slurm&#39;
                                profile.
                                Default: /opt/bactopia/singularity

    --sleep_time            After reading datases, the amount of time (seconds) Nextflow
                                will wait before execution.
                                Default: 5 seconds
Useful Parameters:
    --version               Print workflow version information
    --help                  Show this message and exit
</code></pre></div>

<h2 id="conda-environment">Conda Environment<a class="headerlink" href="#conda-environment" title="Permanent link">&para;</a></h2>
<p>Below is the command that was used to create the Conda environment.
<div class="highlight"><pre><span></span><code>conda create -y -n bactopia-roary -c r -c conda-forge -c bioconda \
    clonalframeml \
    iqtree \
    maskrc-svg \
    ncbi-genome-download \
    pigz \
    prokka \
    r-ggplot2 \
    rename \
    roary \
    snp-dists \
    tbl2asn-forever
</code></pre></div></p>
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p><strong><a href="https://github.com/xavierdidelot/ClonalFrameML">ClonalFramML</a></strong><br />
<em>Didelot, X. &amp; Wilson, D. J. 
<a href="http://dx.doi.org/10.1371/journal.pcbi.1004041">ClonalFrameML: efficient inference of recombination in whole bacterial genomes.</a>
 PLoS Comput. Biol. 11, e1004041 (2015)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Cibiv/IQ-TREE">IQ-TREE</a></strong><br />
<em>L.-T. Nguyen, H.A. Schmidt, A. von Haeseler, B.Q. Minh (2015) 
<a href="https://doi.org/10.1093/molbev/msu300">IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.</a>
 Mol. Biol. Evol., 32:268-274.</em><br />
<em>S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, L.S. Jermiin (2017) 
<a href="https://doi.org/10.1038/nmeth.4285">ModelFinder: Fast model selection for accurate phylogenetic estimates.</a> 
Nat. Methods, 14:587-589.</em><br />
<em>D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, L.S. Vinh (2018) <a href="https://doi.org/10.1093/molbev/msx281">
UFBoot2: Improving the ultrafast bootstrap approximation.</a>
 Mol. Biol. Evol., 35:518–522.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/kwongj/maskrc-svg">maskrc-svg</a></strong><br />
<em>Kwong, J. <a href="https://github.com/kwongj/maskrc-svg">maskrc-svg - Masks recombination as detected by ClonalFrameML or Gubbins and draws an SVG.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a></strong><br />
<em>Blin, K. <a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download: Scripts to download genomes from the NCBI FTP 
servers</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/prokka">Prokka</a></strong><br />
<em>Seemann, T. <a href="http://dx.doi.org/10.1093/bioinformatics/btu153">Prokka: rapid prokaryotic genome annotation</a>. 
Bioinformatics 30, 2068–2069 (2014).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/Roary">Roary</a></strong><br />
<em>Page, A. J. et al. 
<a href="https://doi.org/10.1093/bioinformatics/btv421">Roary: rapid large-scale prokaryote pan genome analysis.</a>
 Bioinformatics 31, 3691–3693 (2015)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/snp-dists">snp-dists</a></strong><br />
<em>Seemann, T. <a href="https://github.com/tseemann/snp-dists">snp-dists - Pairwise SNP distance matrix from a FASTA sequence alignment.</a></em>  </p>
</li>
</ul>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../pirate/" title="pirate" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                pirate
              </div>
            </div>
          </a>
        
        
          <a href="../summary/" title="summary" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                summary
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright © 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/vendor.92ffa368.min.js"></script>
      <script src="../../assets/javascripts/bundle.5123e3d4.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents"}</script>
      
      <script>
        app = initialize({
          base: "../..",
          features: ["tabs"],
          search: Object.assign({
            worker: "../../assets/javascripts/worker/search.a68abb33.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>