



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/bactopia-tools/roary/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.1">
    
    
      
        <title>roary - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.30686662.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700|Ubuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
    
      
        
<script>
  window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-104199698-2", "bactopia.github.io")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })
</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#bactopia-tools-roary" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bactopia
            </span>
            <span class="md-header-nav__topic">
              
                roary
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../.." title="Bactopia" class="md-tabs__link">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../" title="Bactopia Tools" class="md-tabs__link md-tabs__link--active">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../.." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../datasets/" title="Build Datasets" class="md-nav__link">
      Build Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../workflow-overview/" title="Workflow Overview" class="md-nav__link">
      Workflow Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../gtdb/" title="gtdb" class="md-nav__link">
      gtdb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        roary
      </label>
    
    <a href="./" title="roary" class="md-nav__link md-nav__link--active">
      roary
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" title="Example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" title="Output Overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" title="Directory Description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bactopia-info" title="bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clonalframe" title="clonalframe" class="md-nav__link">
    clonalframe
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#iqtree" title="iqtree" class="md-nav__link">
    iqtree
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#refseq" title="refseq" class="md-nav__link">
    refseq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#roary" title="roary" class="md-nav__link">
    roary
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#completed-usage" title="Completed Usage" class="md-nav__link">
    Completed Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example" title="Example" class="md-nav__link">
    Example
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" title="Output Overview" class="md-nav__link">
    Output Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#directory-description" title="Directory Description" class="md-nav__link">
    Directory Description
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bactopia-info" title="bactopia-info" class="md-nav__link">
    bactopia-info
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clonalframe" title="clonalframe" class="md-nav__link">
    clonalframe
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#iqtree" title="iqtree" class="md-nav__link">
    iqtree
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#refseq" title="refseq" class="md-nav__link">
    refseq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#roary" title="roary" class="md-nav__link">
    roary
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#completed-usage" title="Completed Usage" class="md-nav__link">
    Completed Usage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/bactopia-tools/roary.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="bactopia-tools-roary">Bactopia Tools - <em>roary</em><a class="headerlink" href="#bactopia-tools-roary" title="Permanent link">&para;</a></h1>
<p>The <code>roary</code> tool allows you to create a pan-genome of your samples using 
<a href="https://github.com/sanger-pathogens/Roary/">Roary</a>. </p>
<p>Often times, you may
also want to include completed genomes in your pan-genome analysis. This 
is possible with the <code>roary</code> tool. If you use the <code>--species</code> parameter,
all completed genomes available from RefSeq will be downloaded with 
<a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a> and 
reannotated with <a href="https://github.com/tseemann/prokka">Prokka</a> (to make 
compatible gffs). </p>
<p>You can also use the core genome alignment, produced by Roary, to create a 
core genome phylogeny using <a href="https://github.com/xavierdidelot/ClonalFrameML">ClonalFrameMl</a>, 
<a href="https://github.com/kwongj/maskrc-svg">maskrc-svg</a>, and <a href="http://www.iqtree.org/">IQ-TREE</a>.
The core genome pair-wise SNP distance for each sample is also calculated with
<a href="https://github.com/tseemann/snp-dists">snp-dists</a>.</p>
<h2 id="example">Example<a class="headerlink" href="#example" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span>bactopia tools roary \
    --bactopia ~/bactopia-tutorial/bactopia \
    --include ~/bactopia-tutorial/GCF_900475245-include.txt \
    --cpus 4 \
    --n
</pre></div>

<h2 id="output-overview">Output Overview<a class="headerlink" href="#output-overview" title="Permanent link">&para;</a></h2>
<p>Below is the default output structure for the <code>roary</code> tool. Where possible the 
file descriptions below were modified from a tools description.
<div class="highlight"><pre><span></span>roary/
├── bactopia-info
│   ├── roary-report.html
│   ├── roary-timeline.html
│   └── roary-trace.txt
├── clonalframe
│   ├── clonalframe.emsim.txt
│   ├── clonalframe.em.txt
│   ├── clonalframe.importation_status.txt
│   ├── clonalframe.labelled_tree.newick
│   ├── clonalframe.ML_sequence.fasta
│   ├── clonalframe.position_cross_reference.txt
│   ├── core_gene_alignment-masked.aln.gz
│   ├── start-tree.bionj
│   ├── start-tree.ckp.gz
│   ├── start-tree.iqtree
│   ├── start-tree.log
│   ├── start-tree.mldist
│   ├── start-tree.model.gz
│   └── start-tree.treefile
├── core-genome.aligned.fa.gz
├── core-genome.distance.txt
├── core-genome.iqtree
├── iqtree
│   ├── core-genome.alninfo
│   ├── core-genome.bionj
│   ├── core-genome.ckp.gz
│   ├── core-genome.contree
│   ├── core-genome.iqtree
│   ├── core-genome.log
│   ├── core-genome.mldist
│   ├── core-genome.model.gz
│   ├── core-genome.splits.nex
│   ├── core-genome.treefile
│   └── core-genome.ufboot
├── refseq
│   ├── fasta
│   │   └── GCF_900475245.fna
│   └── gff
│       └── GCF_900475245.gff
└── roary
| accessory_binary_genes.fa
| accessory_binary_genes.fa.newick
| accessory_graph.dot
| accessory.header.embl
| accessory.tab
| blast_identity_frequency.Rtab
| clustered_proteins
| conserved_vs_total_genes.png
| core_accessory_graph.dot
| core_accessory.header.embl
| core_accessory.tab
| core_alignment_header.embl
| core_gene_alignment.aln.gz
| gene_presence_absence.csv
| gene_presence_absence.Rtab
| number_of_conserved_genes.Rtab
| number_of_genes_in_pan_genome.Rtab
| number_of_new_genes.Rtab
| number_of_unique_genes.Rtab
| pan_genome_reference.fa
| Rplots.pdf
| summary_statistics.txt
    └── unique_vs_new_genes.png
</pre></div></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>core-genome.aligned.fa.gz</td>
<td>A multiple sequence alignment FASTA of the core genome</td>
</tr>
<tr>
<td>core-genome.distance.txt</td>
<td>Core genome Pair-wise SNP distance for each sample</td>
</tr>
<tr>
<td>core-genome.iqtree</td>
<td>Full result of the IQ-TREE core genome phylogeny</td>
</tr>
</tbody>
</table>
<h3 id="directory-description">Directory Description<a class="headerlink" href="#directory-description" title="Permanent link">&para;</a></h3>
<h4 id="bactopia-info">bactopia-info<a class="headerlink" href="#bactopia-info" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>roary-report.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#execution-report">Execution Report</a></td>
</tr>
<tr>
<td>roary-timeline.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#timeline-report">Timeline Report</a></td>
</tr>
<tr>
<td>roary-trace.txt</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">Trace</a> report</td>
</tr>
</tbody>
</table>
<h4 id="clonalframe">clonalframe<a class="headerlink" href="#clonalframe" title="Permanent link">&para;</a></h4>
<p>Where possible descriptions were taken from the <a href="https://github.com/xavierdidelot/clonalframeml/wiki">ClonalFrameML Wiki</a>, 
IQ-TREE's <a href="https://github.com/Cibiv/IQ-TREE/wiki/Command-Reference">Command Reference</a>
page, <a href="https://github.com/Cibiv/IQ-TREE/wiki/Web-Server-Tutorial">Web Server Tutorial</a> page, 
and the <a href="http://www.iqtree.org/doc/Tutorial">Tutorial</a> page.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>clonalframe.emsim.txt</td>
<td>The bootstrapped values for the three parameters R/theta, nu and delta</td>
</tr>
<tr>
<td>clonalframe.em.txt</td>
<td>The point estimates for R/theta, nu, delta and the branch lengths</td>
</tr>
<tr>
<td>clonalframe.importation_status.txt</td>
<td>The list of reconstructed recombination events</td>
</tr>
<tr>
<td>clonalframe.labelled_tree.newick</td>
<td>The output tree with all nodes labelled so that they can be referred to in other files</td>
</tr>
<tr>
<td>clonalframe.ML_sequence.fasta</td>
<td>The sequence reconstructed by maximum likelihood for all internal nodes of the phylogeny, as well as for all missing data in the input sequences</td>
</tr>
<tr>
<td>clonalframe.position_cross_reference.txt</td>
<td>A vector of comma-separated values indicating for each location in the input sequence file the corresponding position in the sequences in the output <em>ML_sequence.fasta</em> file</td>
</tr>
<tr>
<td>core_gene_alignment-masked.aln.gz</td>
<td>A core-genome alignment with the recomination masked</td>
</tr>
<tr>
<td>start-tree.bionj</td>
<td>A neighbor joining tree produced by BIONJ</td>
</tr>
<tr>
<td>start-tree.ckp.gz</td>
<td>IQ-TREE writes a checkpoint file</td>
</tr>
<tr>
<td>start-tree.iqtree</td>
<td>Full result of the run, this is the main report file</td>
</tr>
<tr>
<td>start-tree.log</td>
<td>Run log</td>
</tr>
<tr>
<td>start-tree.mldist</td>
<td>Contains the likelihood distances</td>
</tr>
<tr>
<td>start-tree.model.gz</td>
<td>Information about all models tested</td>
</tr>
<tr>
<td>start-tree.treefile</td>
<td>Maximum likelihood tree in NEWICK format, can be visualized with treeviewer programs</td>
</tr>
</tbody>
</table>
<h4 id="iqtree">iqtree<a class="headerlink" href="#iqtree" title="Permanent link">&para;</a></h4>
<p>Where possible descriptions were taken from 
IQ-TREE's <a href="https://github.com/Cibiv/IQ-TREE/wiki/Command-Reference">Command Reference</a>
page, <a href="https://github.com/Cibiv/IQ-TREE/wiki/Web-Server-Tutorial">Web Server Tutorial</a> page, 
and the <a href="http://www.iqtree.org/doc/Tutorial">Tutorial</a> page.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>core-genome.alninfo</td>
<td>Alignment site statistics</td>
</tr>
<tr>
<td>core-genome.bionj</td>
<td>A neighbor joining tree produced by BIONJ</td>
</tr>
<tr>
<td>core-genome.ckp.gz</td>
<td>IQ-TREE writes a checkpoint file</td>
</tr>
<tr>
<td>core-genome.contree</td>
<td>Consensus tree with assigned branch supports where branch lengths are optimized on the original alignment; printed if Ultrafast Bootstrap is selected</td>
</tr>
<tr>
<td>core-genome.iqtree</td>
<td>Full result of the run, this is the main report file</td>
</tr>
<tr>
<td>core-genome.log</td>
<td>Run log</td>
</tr>
<tr>
<td>core-genome.mldist</td>
<td>Contains the likelihood distances</td>
</tr>
<tr>
<td>core-genome.model.gz</td>
<td>Information about all models tested</td>
</tr>
<tr>
<td>core-genome.splits.nex</td>
<td>Support values in percentage for all splits (bipartitions), computed as the occurence frequencies in the bootstrap trees</td>
</tr>
<tr>
<td>core-genome.treefile</td>
<td>Maximum likelihood tree in NEWICK format, can be visualized with treeviewer programs</td>
</tr>
<tr>
<td>core-genome.ufboot</td>
<td>Trees created during the bootstrap steps</td>
</tr>
</tbody>
</table>
<h4 id="refseq">refseq<a class="headerlink" href="#refseq" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.fna</td>
<td>FASTA formated genome downloaded from NCBI Assembly database.</td>
</tr>
<tr>
<td>.gff</td>
<td>GFF output from the Prokka re-annotation of the reference FASTA</td>
</tr>
</tbody>
</table>
<h4 id="roary">roary<a class="headerlink" href="#roary" title="Permanent link">&para;</a></h4>
<p>Where possible descriptions were taken from the <a href="http://sanger-pathogens.github.io/Roary/">Roary Documentation</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>accessory_binary_genes.fa</td>
<td>A FASTA file with binary presence and absence of accessory genes</td>
</tr>
<tr>
<td>accessory_binary_genes.fa.newick</td>
<td>A tree created using the binary presence and absence of accessory genes</td>
</tr>
<tr>
<td>accessory_graph.dot</td>
<td>A graph in DOT format of how genes are linked together at the contig level in the accessory genome</td>
</tr>
<tr>
<td>accessory.header.embl</td>
<td>Tab/EMBL formatted file of accessory genes</td>
</tr>
<tr>
<td>accessory.tab</td>
<td>Tab/EMBL formatted file of accessory genes</td>
</tr>
<tr>
<td>blast_identity_frequency.Rtab</td>
<td>Blast results for percentage idenity graph</td>
</tr>
<tr>
<td>clustered_proteins</td>
<td>Groups file where each line lists the sequences in a cluster</td>
</tr>
<tr>
<td>conserved_vs_total_genes.png</td>
<td>Plot compairing conserved genes and total genes</td>
</tr>
<tr>
<td>core_accessory_graph.dot</td>
<td>A graph in DOT format of how genes are linked together at the contig level in the pan genome</td>
</tr>
<tr>
<td>core_accessory.header.embl</td>
<td>Tab/EMBL formatted file of core genes</td>
</tr>
<tr>
<td>core_accessory.tab</td>
<td>Tab/EMBL formatted file of core genes</td>
</tr>
<tr>
<td>core_alignment_header.embl</td>
<td>Tab/EMBL formatted file of core genome alignment</td>
</tr>
<tr>
<td>core_gene_alignment.aln.gz</td>
<td>A multi-FASTA alignment of all of the core genes</td>
</tr>
<tr>
<td>gene_presence_absence.csv</td>
<td>Lists each gene and which samples it is present in</td>
</tr>
<tr>
<td>gene_presence_absence.Rtab</td>
<td>Tab delimited binary matrix with the presence and absence of each gene in each sample</td>
</tr>
<tr>
<td>number_of_conserved_genes.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>number_of_genes_in_pan_genome.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>number_of_new_genes.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>number_of_unique_genes.Rtab</td>
<td>Graphs on how the pan genome varies as genomes are added (in random orders)</td>
</tr>
<tr>
<td>pan_genome_reference.fa</td>
<td>FASTA file which contains a single representative nucleotide sequence from each of the clusters in the pan genome (core and accessory)</td>
</tr>
<tr>
<td>Rplots.pdf</td>
<td>PDF containing each plot</td>
</tr>
<tr>
<td>summary_statistics.txt</td>
<td>Number of genes in the core and accessory</td>
</tr>
<tr>
<td>unique_vs_new_genes.png</td>
<td>Plot compairing new vs old genes</td>
</tr>
</tbody>
</table>
<h2 id="completed-usage">Completed Usage<a class="headerlink" href="#completed-usage" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span>Required Parameters:
    --bactopia STR          Directory containing Bactopia analysis results for all samples.

Optional Parameters:
    --include STR           A text file containing sample names to include in the
                                analysis. The expected format is a single sample per line.

    --exclude STR           A text file containing sample names to exclude from the
                                analysis. The expected format is a single sample per line.

    --prefix DIR            Prefix to use for final output files
                                Default: core-genome

    --outdir DIR            Directory to write results to
                                Default: ./

    --max_time INT          The maximum number of minutes a job should run before being halted.
                                Default: 2880 minutes

    --max_memory INT        The maximum amount of memory (Gb) allowed to a single process.
                                Default: 64 Gb

    --cpus INT              Number of processors made available to a single
                                process.
                                Default: 1

RefSeq Assemblies Related Parameters:
    --species STR           The name of the species to download RefSeq assemblies for. This
                                is a completely optional step and is meant to supplement
                                your dataset with high-quality completed genomes.

    --prokka_evalue STR     Similarity e-value cut-off
                                Default: 1e-09

    --prokka_coverage INT   Minimum coverage on query protein
                                 Default: 80

Roary Related Parameters:
    --o STR                 Clusters output filename
                                Default: clustered_proteins

    --n                     Execute a fast core gene alignment with MAFFT
                                Default: Use PRANK

    --i INT                 Minimum percentage identity for blastp
                                Default: 95

    --cd INT                Percentage of isolates a gene must be in to be core
                                Default: 99%

    --g INT                 Maximum number of clusters
                                Default: 50000

    --s                     Do not split paralogs
                                Default: false

    --ap                    Allow paralogs in core alignment
                                Default: false

    --iv STR                Change the MCL inflation value
                                Default: 1.5

IQ-TREE Related Parameters:
    --skip_phylogeny        Skip the creation a core-genome based phylogeny

    --m STR                 Substitution model name
                                Default: MFP

    --bb INT                Ultrafast bootstrap replicates
                                Default: 1000

    --alrt INT              SH-like approximate likelihood ratio test replicates
                                Default: 1000

    --asr                   Ancestral state reconstruction by empirical Bayes
                                Default: false

    --iqtree_opts STR       Extra IQ-TREE options in quotes.
                                Default: &#39;&#39;

ClonalFrameML Related Parameters:
    --skip_clonalframe      Skip the ClonalFrameML and use the original core-genome
                                alignment for the final tree.

    --emsim INT             Number of simulations to estimate uncertainty in the EM results.
                                Default: 100

    --clonal_opts STR       Extra ClonalFrameML options in quotes.
                                Default: &#39;&#39;

SNP-Dists Related Parameters:
    --a                     Count all differences not just [AGTC]
                                Default: false

    --b                     Blank top left corner cell
                                Default: false

    --c                     Output CSV instead of TSV
                                Default: false

    --k                     Keep case, don&#39;t uppercase all letters
                                Default: false

Nextflow Related Parameters:
    --publish_mode          Set Nextflow&#39;s method for publishing output files. Allowed methods are:
                                &#39;copy&#39; (default)    Copies the output files into the published directory.

                                &#39;copyNoFollow&#39; Copies the output files into the published directory
                                               without following symlinks ie. copies the links themselves.

                                &#39;link&#39;    Creates a hard link in the published directory for each
                                          process output file.

                                &#39;rellink&#39; Creates a relative symbolic link in the published directory
                                          for each process output file.

                                &#39;symlink&#39; Creates an absolute symbolic link in the published directory
                                          for each process output file.

                                Default: copy

    --force                 Nextflow will overwrite existing output files.
                                Default: null

    --conatainerPath        Path to Singularity containers to be used by the &#39;slurm&#39;
                                profile.
                                Default: /opt/bactopia/singularity

    --sleep_time            After reading datases, the amount of time (seconds) Nextflow
                                will wait before execution.
                                Default: 5 seconds
Useful Parameters:
    --version               Print workflow version information
    --help                  Show this message and exit
</pre></div>

<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p><strong><a href="https://github.com/xavierdidelot/ClonalFrameML">ClonalFramML</a></strong><br />
<em>Didelot, X. &amp; Wilson, D. J. 
<a href="http://dx.doi.org/10.1371/journal.pcbi.1004041">ClonalFrameML: efficient inference of recombination in whole bacterial genomes.</a>
 PLoS Comput. Biol. 11, e1004041 (2015)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Cibiv/IQ-TREE">IQ-TREE</a></strong><br />
<em>L.-T. Nguyen, H.A. Schmidt, A. von Haeseler, B.Q. Minh (2015) 
<a href="https://doi.org/10.1093/molbev/msu300">IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.</a>
 Mol. Biol. Evol., 32:268-274.</em><br />
<em>S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, L.S. Jermiin (2017) 
<a href="https://doi.org/10.1038/nmeth.4285">ModelFinder: Fast model selection for accurate phylogenetic estimates.</a> 
Nat. Methods, 14:587-589.</em><br />
<em>D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, L.S. Vinh (2018) <a href="https://doi.org/10.1093/molbev/msx281">
UFBoot2: Improving the ultrafast bootstrap approximation.</a>
 Mol. Biol. Evol., 35:518–522.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/kwongj/maskrc-svg">maskrc-svg</a></strong><br />
<em>Kwong, J. <a href="https://github.com/kwongj/maskrc-svg">maskrc-svg - Masks recombination as detected by ClonalFrameML or Gubbins and draws an SVG.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a></strong><br />
<em>Blin, K. <a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download: Scripts to download genomes from the NCBI FTP 
servers</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/prokka">Prokka</a></strong><br />
<em>Seemann, T. <a href="http://dx.doi.org/10.1093/bioinformatics/btu153">Prokka: rapid prokaryotic genome annotation</a>. 
Bioinformatics 30, 2068–2069 (2014).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/Roary">Roary</a></strong><br />
<em>Page, A. J. et al. 
<a href="https://doi.org/10.1093/bioinformatics/btv421">Roary: rapid large-scale prokaryote pan genome analysis.</a>
 Bioinformatics 31, 3691–3693 (2015)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/snp-dists">snp-dists</a></strong><br />
<em>Seemann, T. <a href="https://github.com/tseemann/snp-dists">snp-dists - Pairwise SNP distance matrix from a FASTA sequence alignment.</a></em>  </p>
</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../phyloflash/" title="phyloflash" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                phyloflash
              </span>
            </div>
          </a>
        
        
          <a href="../summary/" title="summary" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                summary
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright © 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.c648116f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
  </body>
</html>